// @bryntum/scheduler-vue-3/lib/chunks/BfgLmELe.js
import { defineComponent as f, ref as a, provide as g, onMounted as y, getCurrentInstance as F, toRaw as b, onBeforeUnmount as w, openBlock as x, createElementBlock as C, watch as v } from "vue";
import { FieldFilterPickerGroup as A } from "@bryntum/scheduler";
import { WrapperHelper as l } from "../helper/WrapperHelper.js";
const S = /* @__PURE__ */ f({
  __name: "BryntumFieldFilterPickerGroup",
  props: {
    addFilterButtonText: {},
    adopt: {},
    align: {},
    alignSelf: {},
    anchor: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    autoUpdateRecord: {},
    bubbleEvents: {},
    callOnFunctions: {},
    canDeleteFilter: { type: Function },
    canManageFilter: { type: Function },
    centered: {},
    cls: {},
    config: {},
    constrainTo: {},
    content: {},
    contentElementCls: {},
    dataset: {},
    dateFormat: {},
    defaultBindProperty: {},
    defaultFocus: { type: Function },
    defaults: {},
    disabled: {},
    dock: {},
    draggable: {},
    extraData: {},
    fields: {},
    filters: {},
    flex: {},
    floating: {},
    getFieldFilterPickerConfig: { type: Function },
    height: {},
    hidden: {},
    hideAnimation: {},
    hideWhenEmpty: {},
    html: { type: [String, Function, Array] },
    htmlCls: {},
    id: {},
    ignoreParentReadOnly: {},
    insertBefore: {},
    insertFirst: {},
    itemCls: {},
    items: {},
    keyMap: {},
    layout: {},
    layoutStyle: {},
    lazyItems: {},
    limitToProperty: {},
    listeners: {},
    localeClass: {},
    localizable: {},
    localizableProperties: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxHeight: {},
    maximizeOnMobile: {},
    maxWidth: {},
    minHeight: {},
    minWidth: {},
    monitorResize: {},
    namedItems: {},
    operators: {},
    owner: {},
    positioned: {},
    preventTooltipOnTouch: {},
    readOnly: {},
    record: {},
    relayStoreEvents: {},
    ripple: {},
    rootElement: {},
    scrollable: {},
    scrollAction: {},
    showAddFilterButton: {},
    showAnimation: {},
    showTooltipWhenDisabled: {},
    store: {},
    strictRecordMapping: {},
    tab: {},
    tag: {},
    textAlign: {},
    textContent: {},
    title: {},
    tooltip: {},
    triggerChangeOnInput: {},
    type: {},
    ui: {},
    weight: {},
    width: {},
    x: {},
    y: {}
  },
  emits: ["beforeAddFilter", "beforeDestroy", "beforeHide", "beforeSetRecord", "beforeShow", "catchAll", "change", "destroy", "focusIn", "focusOut", "hide", "paint", "recompose", "resize", "show"],
  setup(s, { expose: c, emit: d }) {
    const p = s, m = d, n = {
      instanceClass: A,
      instanceName: "FieldFilterPickerGroup",
      configNames: [
        "addFilterButtonText",
        "adopt",
        "align",
        "anchor",
        "ariaDescription",
        "ariaLabel",
        "autoUpdateRecord",
        "bubbleEvents",
        "callOnFunctions",
        "canDeleteFilter",
        "canManageFilter",
        "centered",
        "config",
        "constrainTo",
        "contentElementCls",
        "dateFormat",
        "defaultBindProperty",
        "defaultFocus",
        "defaults",
        "dock",
        "draggable",
        "fields",
        "filters",
        "floating",
        "getFieldFilterPickerConfig",
        "hideAnimation",
        "hideWhenEmpty",
        "htmlCls",
        "ignoreParentReadOnly",
        "itemCls",
        "lazyItems",
        "limitToProperty",
        "listeners",
        "localeClass",
        "localizable",
        "localizableProperties",
        "maskDefaults",
        "masked",
        "monitorResize",
        "namedItems",
        "operators",
        "owner",
        "positioned",
        "preventTooltipOnTouch",
        "relayStoreEvents",
        "ripple",
        "rootElement",
        "scrollAction",
        "showAddFilterButton",
        "showAnimation",
        "showTooltipWhenDisabled",
        "store",
        "tab",
        "tag",
        "textAlign",
        "textContent",
        "title",
        "triggerChangeOnInput",
        "type",
        "ui",
        "weight"
      ],
      propertyConfigNames: [
        "alignSelf",
        "appendTo",
        "ariaLive",
        "cls",
        "content",
        "dataset",
        "disabled",
        "extraData",
        "flex",
        "height",
        "hidden",
        "html",
        "id",
        "insertBefore",
        "insertFirst",
        "items",
        "keyMap",
        "layout",
        "layoutStyle",
        "margin",
        "maxHeight",
        "maximizeOnMobile",
        "maxWidth",
        "minHeight",
        "minWidth",
        "onBeforeAddFilter",
        "onBeforeDestroy",
        "onBeforeHide",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onCatchAll",
        "onChange",
        "onDestroy",
        "onFocusIn",
        "onFocusOut",
        "onHide",
        "onPaint",
        "onReadOnly",
        "onRecompose",
        "onResize",
        "onShow",
        "readOnly",
        "record",
        "scrollable",
        "strictRecordMapping",
        "tooltip",
        "width",
        "x",
        "y"
      ],
      propertyNames: [
        "accessibility",
        "anchorSize",
        "isSettingValues",
        "isValid",
        "value",
        "values"
      ],
      eventNames: [
        "beforeAddFilter",
        "beforeDestroy",
        "beforeHide",
        "beforeSetRecord",
        "beforeShow",
        "catchAll",
        "change",
        "destroy",
        "focusIn",
        "focusOut",
        "hide",
        "paint",
        "recompose",
        "resize",
        "show"
      ]
    }, o = {}, i = a(), r = a(/* @__PURE__ */ new Map());
    return g("teleports", r), y(() => {
      const e = F();
      o.value = l.createWidget({
        me: e,
        props: p,
        emit: m,
        widgetData: n,
        element: i.value,
        toRaw: b
      });
      const t = (u, h) => v(() => e.props[u], h);
      l.watchProps(o.value, n, t);
    }), w(() => {
      var e, t;
      (t = (e = o.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), c({
      instance: o,
      refElement: i,
      teleports: r
    }), (e, t) => (x(), C("div", {
      ref_key: "refElement",
      ref: i
    }, null, 512));
  }
});
export {
  S as _
};
