// @bryntum/scheduler-vue-3/lib/chunks/CwvE4CGw.js
import { defineComponent as f, ref as l, provide as g, onMounted as y, getCurrentInstance as b, toRaw as C, onBeforeUnmount as k, openBlock as D, createElementBlock as w, watch as S } from "vue";
import { DatePicker as R } from "@bryntum/scheduler";
import { WrapperHelper as s } from "../helper/WrapperHelper.js";
const F = /* @__PURE__ */ f({
  __name: "BryntumDatePicker",
  props: {
    activeDate: {},
    adopt: {},
    align: {},
    alignSelf: {},
    anchor: {},
    animateTimeShift: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    autoUpdateRecord: {},
    bbar: {},
    bodyCls: {},
    bubbleEvents: {},
    callOnFunctions: {},
    cellRenderer: { type: [Function, String] },
    centered: {},
    cls: {},
    collapsed: {},
    collapsible: {},
    config: {},
    constrainTo: {},
    content: {},
    contentElementCls: {},
    dataset: {},
    date: {},
    dayNameFormat: {},
    defaultBindProperty: {},
    defaultFocus: { type: Function },
    defaults: {},
    disabled: {},
    disabledCls: {},
    disabledDates: { type: [Function, Array, String] },
    disableNonWorkingDays: {},
    disableOtherMonthCells: {},
    disableWeekends: {},
    dock: {},
    draggable: {},
    editMonth: {},
    extraData: {},
    flex: {},
    floating: {},
    focusDisabledDates: {},
    footer: {},
    header: {},
    headerRenderer: { type: [Function, String] },
    height: {},
    hidden: {},
    hideAnimation: {},
    hideOtherMonthCells: {},
    hideWhenEmpty: {},
    highlightSelectedWeek: {},
    html: { type: [String, Function, Array] },
    htmlCls: {},
    id: {},
    ignoreParentReadOnly: {},
    includeYear: {},
    insertBefore: {},
    insertFirst: {},
    itemCls: {},
    items: {},
    keyMap: {},
    layout: {},
    layoutStyle: {},
    lazyItems: {},
    listeners: {},
    localeClass: {},
    localizable: {},
    localizableProperties: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxDate: {},
    maxHeight: {},
    maximizeOnMobile: {},
    maxWidth: {},
    minColumnWidth: {},
    minDate: {},
    minHeight: {},
    minRowHeight: {},
    minWidth: {},
    monitorResize: {},
    month: {},
    multiSelect: {},
    namedItems: {},
    nonWorkingDayCls: {},
    nonWorkingDays: {},
    otherMonthCls: {},
    owner: {},
    positioned: {},
    preventTooltipOnTouch: {},
    readOnly: {},
    record: {},
    relayStoreEvents: {},
    ripple: {},
    rootElement: {},
    scrollable: {},
    scrollAction: {},
    selection: {},
    showAnimation: {},
    showTooltipWhenDisabled: {},
    showWeekColumn: {},
    sixWeeks: {},
    stateful: {},
    statefulEvents: {},
    stateId: {},
    stateProvider: {},
    strictRecordMapping: {},
    strips: {},
    tab: {},
    tag: {},
    tbar: {},
    textAlign: {},
    textContent: {},
    tip: {},
    title: {},
    todayCls: {},
    tools: {},
    tooltip: {},
    trapFocus: {},
    type: {},
    ui: {},
    weekendCls: {},
    weekRenderer: { type: [Function, String] },
    weekStartDay: {},
    weight: {},
    width: {},
    x: {},
    y: {}
  },
  emits: ["beforeDestroy", "beforeHide", "beforeRefresh", "beforeSetRecord", "beforeShow", "catchAll", "collapse", "dateChange", "destroy", "expand", "focusIn", "focusOut", "hide", "paint", "recompose", "refresh", "resize", "selectionChange", "show", "toolClick"],
  setup(r, { expose: d, emit: c }) {
    const h = r, m = c, a = {
      instanceClass: R,
      instanceName: "DatePicker",
      configNames: [
        "activeDate",
        "adopt",
        "align",
        "anchor",
        "ariaDescription",
        "ariaLabel",
        "autoUpdateRecord",
        "bbar",
        "bodyCls",
        "bubbleEvents",
        "callOnFunctions",
        "cellRenderer",
        "centered",
        "collapsible",
        "config",
        "constrainTo",
        "contentElementCls",
        "dayNameFormat",
        "defaultBindProperty",
        "defaultFocus",
        "defaults",
        "disabledCls",
        "disabledDates",
        "disableNonWorkingDays",
        "disableOtherMonthCells",
        "disableWeekends",
        "dock",
        "draggable",
        "editMonth",
        "floating",
        "focusDisabledDates",
        "footer",
        "header",
        "headerRenderer",
        "hideAnimation",
        "hideOtherMonthCells",
        "hideWhenEmpty",
        "htmlCls",
        "ignoreParentReadOnly",
        "includeYear",
        "itemCls",
        "lazyItems",
        "listeners",
        "localeClass",
        "localizable",
        "localizableProperties",
        "maskDefaults",
        "masked",
        "maxDate",
        "minColumnWidth",
        "minDate",
        "minRowHeight",
        "monitorResize",
        "month",
        "multiSelect",
        "namedItems",
        "nonWorkingDayCls",
        "nonWorkingDays",
        "otherMonthCls",
        "owner",
        "positioned",
        "preventTooltipOnTouch",
        "relayStoreEvents",
        "ripple",
        "rootElement",
        "scrollAction",
        "showAnimation",
        "showTooltipWhenDisabled",
        "showWeekColumn",
        "sixWeeks",
        "stateful",
        "statefulEvents",
        "stateId",
        "stateProvider",
        "strips",
        "tab",
        "tag",
        "tbar",
        "textAlign",
        "textContent",
        "tip",
        "todayCls",
        "trapFocus",
        "type",
        "ui",
        "weekendCls",
        "weekRenderer",
        "weekStartDay",
        "weight"
      ],
      propertyConfigNames: [
        "alignSelf",
        "animateTimeShift",
        "appendTo",
        "ariaLive",
        "cls",
        "collapsed",
        "content",
        "dataset",
        "date",
        "disabled",
        "extraData",
        "flex",
        "height",
        "hidden",
        "highlightSelectedWeek",
        "html",
        "id",
        "insertBefore",
        "insertFirst",
        "items",
        "keyMap",
        "layout",
        "layoutStyle",
        "margin",
        "maxHeight",
        "maximizeOnMobile",
        "maxWidth",
        "minHeight",
        "minWidth",
        "onBeforeDestroy",
        "onBeforeHide",
        "onBeforeRefresh",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onCatchAll",
        "onCollapse",
        "onDateChange",
        "onDestroy",
        "onExpand",
        "onFocusIn",
        "onFocusOut",
        "onHide",
        "onPaint",
        "onReadOnly",
        "onRecompose",
        "onRefresh",
        "onResize",
        "onSelectionChange",
        "onShow",
        "onToolClick",
        "readOnly",
        "record",
        "scrollable",
        "selection",
        "strictRecordMapping",
        "title",
        "tools",
        "tooltip",
        "width",
        "x",
        "y"
      ],
      propertyNames: [
        "accessibility",
        "anchorSize",
        "isSettingValues",
        "isValid",
        "state",
        "values"
      ],
      eventNames: [
        "beforeDestroy",
        "beforeHide",
        "beforeRefresh",
        "beforeSetRecord",
        "beforeShow",
        "catchAll",
        "collapse",
        "dateChange",
        "destroy",
        "expand",
        "focusIn",
        "focusOut",
        "hide",
        "paint",
        "recompose",
        "refresh",
        "resize",
        "selectionChange",
        "show",
        "toolClick"
      ]
    }, o = {}, n = l(), i = l(/* @__PURE__ */ new Map());
    return g("teleports", i), y(() => {
      const e = b();
      o.value = s.createWidget({
        me: e,
        props: h,
        emit: m,
        widgetData: a,
        element: n.value,
        toRaw: C
      });
      const t = (p, u) => S(() => e.props[p], u);
      s.watchProps(o.value, a, t);
    }), k(() => {
      var e, t;
      (t = (e = o.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), d({
      instance: o,
      refElement: n,
      teleports: i
    }), (e, t) => (D(), w("div", {
      ref_key: "refElement",
      ref: n
    }, null, 512));
  }
});
export {
  F as _
};
