// @bryntum/scheduler-vue-3/lib/chunks/D2nukoFP.js
import { defineComponent as h, ref as u, provide as D, onMounted as F, getCurrentInstance as b, toRaw as w, onBeforeUnmount as M, openBlock as a, createElementBlock as d, createElementVNode as S, Fragment as x, renderList as T, createBlock as g, Teleport as E, resolveDynamicComponent as H, mergeProps as k, watch as v } from "vue";
import { TimelineHistogram as y } from "@bryntum/scheduler";
import { WrapperHelper as n } from "../helper/WrapperHelper.js";
const B = {
  ref: "teleportsHolder",
  style: { display: "none" }
}, I = /* @__PURE__ */ h({
  __name: "BryntumTimelineHistogram",
  props: {
    adopt: {},
    aggregateDataEntry: { type: [Function, String] },
    aggregateHistogramDataForGroups: {},
    alignSelf: {},
    animateRemovingRows: {},
    animateTreeNodeToggle: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    autoAdjustTimeAxis: {},
    autoHeight: {},
    barMargin: {},
    barTooltipTemplate: { type: Function },
    bbar: {},
    bodyCls: {},
    bubbleEvents: {},
    bufferCoef: {},
    bufferThreshold: {},
    callOnFunctions: {},
    cls: {},
    collapsed: {},
    collapsible: {},
    columnLines: {},
    columns: {},
    config: {},
    contentElementCls: {},
    contextMenuTriggerEvent: {},
    data: {},
    dataModelField: {},
    dataset: {},
    defaultRegion: {},
    destroyStore: {},
    disabled: {},
    disableGridColumnIdWarning: {},
    disableGridRowModelWarning: {},
    displayDateFormat: {},
    dock: {},
    durationDisplayPrecision: {},
    emptyText: {},
    enableEventAnimations: {},
    enableRecurringEvents: {},
    enableSticky: {},
    enableTextSelection: {},
    enableUndoRedoKeys: {},
    endDate: {},
    eventColor: {},
    eventStyle: {},
    extraData: {},
    fillLastColumn: {},
    fillTicks: {},
    fixedRowHeight: {},
    flex: {},
    footer: {},
    forceFit: {},
    fullRowRefresh: {},
    getBarClass: { type: Function },
    getBarDOMConfig: { type: Function },
    getBarText: { type: Function },
    getBarTextDOMConfig: { type: Function },
    getDataEntryForAggregating: { type: [Function, String] },
    getOutlineClass: { type: Function },
    getOutlineDOMConfig: { type: Function },
    getRecordData: { type: [Function, String] },
    getRowHeight: { type: Function },
    hardRefreshOnTimeAxisReconfigure: {},
    header: {},
    height: {},
    hidden: {},
    hideFooters: {},
    hideHeaders: {},
    hideHorizontalScrollbar: {},
    hideRowHover: {},
    histogramWidget: {},
    histogramWidgetClass: {},
    hoverCls: {},
    id: {},
    ignoreDomEventsWhileScrolling: {},
    ignoreParentReadOnly: {},
    infiniteScroll: {},
    initAggregatedDataEntry: { type: [Function, String] },
    insertBefore: {},
    insertFirst: {},
    keyMap: {},
    listeners: {},
    loadMask: {},
    loadMaskDefaults: {},
    loadMaskError: {},
    localizable: {},
    longPressTime: {},
    managedEventSizing: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxHeight: {},
    maxTimeAxisUnit: {},
    maxWidth: {},
    maxZoomLevel: {},
    minHeight: {},
    minWidth: {},
    minZoomLevel: {},
    monitorResize: {},
    owner: {},
    partner: {},
    plugins: {},
    preserveFocusOnDatasetChange: {},
    preserveScroll: {},
    preserveScrollOnDatasetChange: {},
    presets: {},
    preventTooltipOnTouch: {},
    readOnly: {},
    relayStoreEvents: {},
    resizeToFitIncludesHeader: {},
    resourceTimeRangeRenderer: { type: Function },
    responsiveLevels: {},
    ripple: {},
    rootElement: {},
    rowHeight: {},
    rowLines: {},
    scaleColumn: {},
    scalePoints: {},
    scrollable: {},
    scrollerClass: {},
    scrollManager: {},
    selectionMode: {},
    series: {},
    showBarTip: {},
    showDirty: {},
    snap: {},
    snapRelativeToEventStartDate: {},
    startDate: {},
    stateful: {},
    statefulEvents: {},
    stateId: {},
    stateProvider: {},
    stickyHeaders: {},
    store: {},
    strips: {},
    subGridConfigs: {},
    suppressFit: {},
    syncMask: {},
    tab: {},
    tbar: {},
    tickSize: {},
    timeAxis: {},
    timeZone: {},
    title: {},
    tools: {},
    transitionDuration: {},
    type: {},
    ui: {},
    viewPreset: {},
    visibleDate: {},
    visibleZoomFactor: {},
    weekStartDay: {},
    weight: {},
    width: {},
    workingTime: {},
    zoomKeepsOriginalTimespan: {},
    zoomOnMouseWheel: {},
    zoomOnTimeAxisDoubleClick: {},
    cellCopyPasteFeature: {},
    cellEditFeature: {},
    cellMenuFeature: {},
    cellTooltipFeature: {},
    columnAutoWidthFeature: {},
    columnDragToolbarFeature: {},
    columnLinesFeature: {},
    columnPickerFeature: {},
    columnRenameFeature: {},
    columnReorderFeature: {},
    columnResizeFeature: {},
    excelExporterFeature: {},
    fileDropFeature: {},
    fillHandleFeature: {},
    filterFeature: {},
    filterBarFeature: {},
    groupFeature: {},
    groupSummaryFeature: {},
    headerMenuFeature: {},
    headerZoomFeature: {},
    lockRowsFeature: {},
    mergeCellsFeature: {},
    nonWorkingTimeFeature: {},
    panFeature: {},
    pdfExportFeature: {},
    printFeature: {},
    quickFindFeature: {},
    regionResizeFeature: {},
    resourceMenuFeature: {},
    resourceTimeRangesFeature: {},
    rowCopyPasteFeature: {},
    rowExpanderFeature: {},
    rowReorderFeature: {},
    rowResizeFeature: {},
    scheduleTooltipFeature: {},
    searchFeature: {},
    sortFeature: {},
    splitFeature: {},
    stickyCellsFeature: {},
    stripeFeature: {},
    summaryFeature: {},
    timeAxisHeaderMenuFeature: {},
    timeRangesFeature: {},
    treeFeature: {},
    treeGroupFeature: {}
  },
  emits: ["beforeAssignmentDelete", "beforeCancelCellEdit", "beforeCellDelete", "beforeCellEditStart", "beforeColumnDragStart", "beforeColumnDropFinalize", "beforeColumnResize", "beforeCopy", "beforeCSVExport", "beforeDestroy", "beforeEventDelete", "beforeExcelExport", "beforeFillHandleDragStart", "beforeFinishCellEdit", "beforeHide", "beforeHistogramDataCacheSet", "beforePan", "beforePaste", "beforePdfExport", "beforePresetChange", "beforeRenderHistogramRow", "beforeRenderRecordHistogram", "beforeRenderRow", "beforeRenderRows", "beforeRowCollapse", "beforeRowExpand", "beforeSelectionChange", "beforeSetRecord", "beforeShow", "beforeToggleGroup", "beforeToggleNode", "cancelCellEdit", "catchAll", "cellClick", "cellContextMenu", "cellDblClick", "cellMenuBeforeShow", "cellMenuItem", "cellMenuShow", "cellMenuToggleItem", "cellMouseEnter", "cellMouseLeave", "cellMouseOut", "cellMouseOver", "collapse", "collapseNode", "columnDrag", "columnDragStart", "columnDrop", "columnResize", "columnResizeStart", "contextMenuItem", "contextMenuToggleItem", "copy", "currentTimelineUpdate", "dataChange", "dateRangeChange", "destroy", "dragSelecting", "expand", "expandNode", "fileDrop", "fillHandleBeforeDragFinalize", "fillHandleDrag", "fillHandleDragAbort", "fillHandleDragEnd", "fillHandleDragStart", "finishCellEdit", "focusIn", "focusOut", "gridRowBeforeDragStart", "gridRowBeforeDropFinalize", "gridRowDrag", "gridRowDragAbort", "gridRowDragStart", "gridRowDrop", "headerClick", "headerMenuBeforeShow", "headerMenuItem", "headerMenuShow", "headerMenuToggleItem", "hide", "histogramDataCacheSet", "lockRows", "mouseOut", "mouseOver", "paint", "paste", "pdfExport", "presetChange", "recompose", "renderRow", "renderRows", "resize", "resourceMenuBeforeShow", "resourceMenuItem", "resourceMenuShow", "resourceMenuToggleItem", "resourceTimeRangeClick", "resourceTimeRangeContextMenu", "resourceTimeRangeDblClick", "resourceTimeRangeMouseDown", "resourceTimeRangeMouseOut", "resourceTimeRangeMouseOver", "resourceTimeRangeMouseUp", "responsive", "rowCollapse", "rowExpand", "rowMouseEnter", "rowMouseLeave", "scroll", "selectionChange", "selectionModeChange", "show", "split", "splitterCollapseClick", "splitterDragEnd", "splitterDragStart", "splitterExpandClick", "startCellEdit", "subGridCollapse", "subGridExpand", "timeAxisChange", "timeAxisHeaderClick", "timeAxisHeaderContextMenu", "timeAxisHeaderDblClick", "timeAxisHeaderMenuBeforeShow", "timeAxisHeaderMenuItem", "timeAxisHeaderMenuShow", "timelineContextChange", "timelineViewportResize", "timeRangeHeaderClick", "timeRangeHeaderContextMenu", "timeRangeHeaderDblClick", "toggleGroup", "toggleNode", "toolClick", "unlockRows", "unsplit", "visibleDateRangeChange"],
  setup(c, { expose: m, emit: p }) {
    const f = c, C = p, s = {
      instanceClass: y,
      instanceName: "TimelineHistogram",
      isView: !0,
      dataStores: {
        originalStore: "originals"
      },
      projectStores: !1,
      featureNames: [
        "cellCopyPasteFeature",
        "cellEditFeature",
        "cellMenuFeature",
        "cellTooltipFeature",
        "columnAutoWidthFeature",
        "columnDragToolbarFeature",
        "columnLinesFeature",
        "columnPickerFeature",
        "columnRenameFeature",
        "columnReorderFeature",
        "columnResizeFeature",
        "excelExporterFeature",
        "fileDropFeature",
        "fillHandleFeature",
        "filterFeature",
        "filterBarFeature",
        "groupFeature",
        "groupSummaryFeature",
        "headerMenuFeature",
        "headerZoomFeature",
        "lockRowsFeature",
        "mergeCellsFeature",
        "nonWorkingTimeFeature",
        "panFeature",
        "pdfExportFeature",
        "printFeature",
        "quickFindFeature",
        "regionResizeFeature",
        "resourceMenuFeature",
        "resourceTimeRangesFeature",
        "rowCopyPasteFeature",
        "rowExpanderFeature",
        "rowReorderFeature",
        "rowResizeFeature",
        "scheduleTooltipFeature",
        "searchFeature",
        "sortFeature",
        "splitFeature",
        "stickyCellsFeature",
        "stripeFeature",
        "summaryFeature",
        "timeAxisHeaderMenuFeature",
        "timeRangesFeature",
        "treeFeature",
        "treeGroupFeature"
      ],
      configNames: [
        "adopt",
        "aggregateDataEntry",
        "aggregateHistogramDataForGroups",
        "animateRemovingRows",
        "animateTreeNodeToggle",
        "ariaDescription",
        "ariaLabel",
        "autoAdjustTimeAxis",
        "autoHeight",
        "barTooltipTemplate",
        "bbar",
        "bodyCls",
        "bubbleEvents",
        "bufferCoef",
        "bufferThreshold",
        "callOnFunctions",
        "collapsible",
        "config",
        "contentElementCls",
        "contextMenuTriggerEvent",
        "dataModelField",
        "defaultRegion",
        "destroyStore",
        "disableGridColumnIdWarning",
        "disableGridRowModelWarning",
        "dock",
        "durationDisplayPrecision",
        "enableRecurringEvents",
        "enableSticky",
        "enableTextSelection",
        "fillLastColumn",
        "fixedRowHeight",
        "footer",
        "fullRowRefresh",
        "getBarClass",
        "getBarDOMConfig",
        "getBarText",
        "getBarTextDOMConfig",
        "getDataEntryForAggregating",
        "getOutlineClass",
        "getOutlineDOMConfig",
        "getRecordData",
        "getRowHeight",
        "hardRefreshOnTimeAxisReconfigure",
        "header",
        "hideHorizontalScrollbar",
        "histogramWidgetClass",
        "hoverCls",
        "ignoreDomEventsWhileScrolling",
        "ignoreParentReadOnly",
        "infiniteScroll",
        "initAggregatedDataEntry",
        "listeners",
        "loadMask",
        "loadMaskDefaults",
        "loadMaskError",
        "localizable",
        "managedEventSizing",
        "maskDefaults",
        "masked",
        "maxTimeAxisUnit",
        "monitorResize",
        "owner",
        "partner",
        "plugins",
        "preserveFocusOnDatasetChange",
        "preserveScrollOnDatasetChange",
        "preventTooltipOnTouch",
        "relayStoreEvents",
        "resizeToFitIncludesHeader",
        "resourceTimeRangeRenderer",
        "responsiveLevels",
        "ripple",
        "rootElement",
        "scaleColumn",
        "scalePoints",
        "scrollerClass",
        "scrollManager",
        "series",
        "showDirty",
        "snapRelativeToEventStartDate",
        "stateful",
        "statefulEvents",
        "stateId",
        "stateProvider",
        "stickyHeaders",
        "strips",
        "subGridConfigs",
        "suppressFit",
        "syncMask",
        "tab",
        "tbar",
        "timeAxis",
        "type",
        "ui",
        "visibleZoomFactor",
        "weekStartDay",
        "weight",
        "zoomKeepsOriginalTimespan",
        "zoomOnMouseWheel",
        "zoomOnTimeAxisDoubleClick"
      ],
      propertyConfigNames: [
        "alignSelf",
        "appendTo",
        "ariaLive",
        "barMargin",
        "cls",
        "collapsed",
        "columnLines",
        "columns",
        "data",
        "dataset",
        "disabled",
        "displayDateFormat",
        "emptyText",
        "enableEventAnimations",
        "enableUndoRedoKeys",
        "endDate",
        "eventColor",
        "eventStyle",
        "extraData",
        "fillTicks",
        "flex",
        "forceFit",
        "height",
        "hidden",
        "hideFooters",
        "hideHeaders",
        "hideRowHover",
        "histogramWidget",
        "id",
        "insertBefore",
        "insertFirst",
        "keyMap",
        "longPressTime",
        "margin",
        "maxHeight",
        "maxWidth",
        "maxZoomLevel",
        "minHeight",
        "minWidth",
        "minZoomLevel",
        "onBeforeAssignmentDelete",
        "onBeforeCancelCellEdit",
        "onBeforeCellDelete",
        "onBeforeCellEditStart",
        "onBeforeColumnDragStart",
        "onBeforeColumnDropFinalize",
        "onBeforeColumnResize",
        "onBeforeCopy",
        "onBeforeCSVExport",
        "onBeforeDestroy",
        "onBeforeEventDelete",
        "onBeforeExcelExport",
        "onBeforeFillHandleDragStart",
        "onBeforeFinishCellEdit",
        "onBeforeHide",
        "onBeforeHistogramDataCacheSet",
        "onBeforePan",
        "onBeforePaste",
        "onBeforePdfExport",
        "onBeforePresetChange",
        "onBeforeRenderHistogramRow",
        "onBeforeRenderRecordHistogram",
        "onBeforeRenderRow",
        "onBeforeRenderRows",
        "onBeforeRowCollapse",
        "onBeforeRowExpand",
        "onBeforeSelectionChange",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onBeforeToggleGroup",
        "onBeforeToggleNode",
        "onCancelCellEdit",
        "onCatchAll",
        "onCellClick",
        "onCellContextMenu",
        "onCellDblClick",
        "onCellMenuBeforeShow",
        "onCellMenuItem",
        "onCellMenuShow",
        "onCellMenuToggleItem",
        "onCellMouseEnter",
        "onCellMouseLeave",
        "onCellMouseOut",
        "onCellMouseOver",
        "onCollapse",
        "onCollapseNode",
        "onColumnDrag",
        "onColumnDragStart",
        "onColumnDrop",
        "onColumnResize",
        "onColumnResizeStart",
        "onContextMenuItem",
        "onContextMenuToggleItem",
        "onCopy",
        "onCurrentTimelineUpdate",
        "onDataChange",
        "onDateRangeChange",
        "onDestroy",
        "onDragSelecting",
        "onExpand",
        "onExpandNode",
        "onFileDrop",
        "onFillHandleBeforeDragFinalize",
        "onFillHandleDrag",
        "onFillHandleDragAbort",
        "onFillHandleDragEnd",
        "onFillHandleDragStart",
        "onFinishCellEdit",
        "onFocusIn",
        "onFocusOut",
        "onGridRowBeforeDragStart",
        "onGridRowBeforeDropFinalize",
        "onGridRowDrag",
        "onGridRowDragAbort",
        "onGridRowDragStart",
        "onGridRowDrop",
        "onHeaderClick",
        "onHeaderMenuBeforeShow",
        "onHeaderMenuItem",
        "onHeaderMenuShow",
        "onHeaderMenuToggleItem",
        "onHide",
        "onHistogramDataCacheSet",
        "onLockRows",
        "onMouseOut",
        "onMouseOver",
        "onPaint",
        "onPaste",
        "onPdfExport",
        "onPresetChange",
        "onReadOnly",
        "onRecompose",
        "onRenderRow",
        "onRenderRows",
        "onResize",
        "onResourceMenuBeforeShow",
        "onResourceMenuItem",
        "onResourceMenuShow",
        "onResourceMenuToggleItem",
        "onResourceTimeRangeClick",
        "onResourceTimeRangeContextMenu",
        "onResourceTimeRangeDblClick",
        "onResourceTimeRangeMouseDown",
        "onResourceTimeRangeMouseOut",
        "onResourceTimeRangeMouseOver",
        "onResourceTimeRangeMouseUp",
        "onResponsive",
        "onRowCollapse",
        "onRowExpand",
        "onRowMouseEnter",
        "onRowMouseLeave",
        "onScroll",
        "onSelectionChange",
        "onSelectionModeChange",
        "onShow",
        "onSplit",
        "onSplitterCollapseClick",
        "onSplitterDragEnd",
        "onSplitterDragStart",
        "onSplitterExpandClick",
        "onStartCellEdit",
        "onSubGridCollapse",
        "onSubGridExpand",
        "onTimeAxisChange",
        "onTimeAxisHeaderClick",
        "onTimeAxisHeaderContextMenu",
        "onTimeAxisHeaderDblClick",
        "onTimeAxisHeaderMenuBeforeShow",
        "onTimeAxisHeaderMenuItem",
        "onTimeAxisHeaderMenuShow",
        "onTimelineContextChange",
        "onTimelineViewportResize",
        "onTimeRangeHeaderClick",
        "onTimeRangeHeaderContextMenu",
        "onTimeRangeHeaderDblClick",
        "onToggleGroup",
        "onToggleNode",
        "onToolClick",
        "onUnlockRows",
        "onUnsplit",
        "onVisibleDateRangeChange",
        "preserveScroll",
        "presets",
        "readOnly",
        "rowHeight",
        "rowLines",
        "scrollable",
        "selectionMode",
        "showBarTip",
        "snap",
        "startDate",
        "store",
        "tickSize",
        "timeZone",
        "title",
        "tools",
        "transitionDuration",
        "viewPreset",
        "visibleDate",
        "width",
        "workingTime"
      ],
      propertyNames: [
        "accessibility",
        "originalStore",
        "scrollLeft",
        "scrollTop",
        "scrollX",
        "selectedCell",
        "selectedCells",
        "selectedRecord",
        "selectedRecords",
        "selectedRows",
        "state",
        "timeResolution",
        "tooltip",
        "zoomLevel"
      ],
      eventNames: [
        "beforeAssignmentDelete",
        "beforeCancelCellEdit",
        "beforeCellDelete",
        "beforeCellEditStart",
        "beforeColumnDragStart",
        "beforeColumnDropFinalize",
        "beforeColumnResize",
        "beforeCopy",
        "beforeCSVExport",
        "beforeDestroy",
        "beforeEventDelete",
        "beforeExcelExport",
        "beforeFillHandleDragStart",
        "beforeFinishCellEdit",
        "beforeHide",
        "beforeHistogramDataCacheSet",
        "beforePan",
        "beforePaste",
        "beforePdfExport",
        "beforePresetChange",
        "beforeRenderHistogramRow",
        "beforeRenderRecordHistogram",
        "beforeRenderRow",
        "beforeRenderRows",
        "beforeRowCollapse",
        "beforeRowExpand",
        "beforeSelectionChange",
        "beforeSetRecord",
        "beforeShow",
        "beforeToggleGroup",
        "beforeToggleNode",
        "cancelCellEdit",
        "catchAll",
        "cellClick",
        "cellContextMenu",
        "cellDblClick",
        "cellMenuBeforeShow",
        "cellMenuItem",
        "cellMenuShow",
        "cellMenuToggleItem",
        "cellMouseEnter",
        "cellMouseLeave",
        "cellMouseOut",
        "cellMouseOver",
        "collapse",
        "collapseNode",
        "columnDrag",
        "columnDragStart",
        "columnDrop",
        "columnResize",
        "columnResizeStart",
        "contextMenuItem",
        "contextMenuToggleItem",
        "copy",
        "currentTimelineUpdate",
        "dataChange",
        "dateRangeChange",
        "destroy",
        "dragSelecting",
        "expand",
        "expandNode",
        "fileDrop",
        "fillHandleBeforeDragFinalize",
        "fillHandleDrag",
        "fillHandleDragAbort",
        "fillHandleDragEnd",
        "fillHandleDragStart",
        "finishCellEdit",
        "focusIn",
        "focusOut",
        "gridRowBeforeDragStart",
        "gridRowBeforeDropFinalize",
        "gridRowDrag",
        "gridRowDragAbort",
        "gridRowDragStart",
        "gridRowDrop",
        "headerClick",
        "headerMenuBeforeShow",
        "headerMenuItem",
        "headerMenuShow",
        "headerMenuToggleItem",
        "hide",
        "histogramDataCacheSet",
        "lockRows",
        "mouseOut",
        "mouseOver",
        "paint",
        "paste",
        "pdfExport",
        "presetChange",
        "recompose",
        "renderRow",
        "renderRows",
        "resize",
        "resourceMenuBeforeShow",
        "resourceMenuItem",
        "resourceMenuShow",
        "resourceMenuToggleItem",
        "resourceTimeRangeClick",
        "resourceTimeRangeContextMenu",
        "resourceTimeRangeDblClick",
        "resourceTimeRangeMouseDown",
        "resourceTimeRangeMouseOut",
        "resourceTimeRangeMouseOver",
        "resourceTimeRangeMouseUp",
        "responsive",
        "rowCollapse",
        "rowExpand",
        "rowMouseEnter",
        "rowMouseLeave",
        "scroll",
        "selectionChange",
        "selectionModeChange",
        "show",
        "split",
        "splitterCollapseClick",
        "splitterDragEnd",
        "splitterDragStart",
        "splitterExpandClick",
        "startCellEdit",
        "subGridCollapse",
        "subGridExpand",
        "timeAxisChange",
        "timeAxisHeaderClick",
        "timeAxisHeaderContextMenu",
        "timeAxisHeaderDblClick",
        "timeAxisHeaderMenuBeforeShow",
        "timeAxisHeaderMenuItem",
        "timeAxisHeaderMenuShow",
        "timelineContextChange",
        "timelineViewportResize",
        "timeRangeHeaderClick",
        "timeRangeHeaderContextMenu",
        "timeRangeHeaderDblClick",
        "toggleGroup",
        "toggleNode",
        "toolClick",
        "unlockRows",
        "unsplit",
        "visibleDateRangeChange"
      ]
    }, r = {}, l = u(), i = u(/* @__PURE__ */ new Map());
    return D("teleports", i), F(() => {
      const e = b();
      r.value = n.createWidget({
        me: e,
        props: f,
        emit: C,
        widgetData: s,
        element: l.value,
        processCellContent: n.processCellContent,
        hasFrameworkRenderer: n.hasFrameworkRenderer,
        toRaw: w
      }), n.relayStores(e.data, r.value);
      const t = (o, R) => v(() => e.props[o], R);
      n.watchProps(r.value, s, t);
    }), M(() => {
      var e, t;
      (t = (e = r.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), m({
      instance: r,
      refElement: l,
      teleports: i
    }), (e, t) => (a(), d("div", {
      ref_key: "refElement",
      ref: l
    }, [
      S("div", B, [
        (a(!0), d(x, null, T(i.value.values(), (o) => (a(), g(E, {
          key: o.key,
          to: o.to
        }, [
          (a(), g(H(o.is), k({
            ref_for: !0,
            ref: o.key,
            "data-key": o.key,
            "data-generation": o.generation
          }, o.bind, { "data-vue": "" }), null, 16, ["data-key", "data-generation"]))
        ], 8, ["to"]))), 128))
      ], 512)
    ], 512));
  }
});
export {
  I as _
};
