// @bryntum/scheduler-vue-3/lib/chunks/DJ5NC6Bm.js
import { defineComponent as C, ref as r, provide as S, onMounted as D, getCurrentInstance as R, toRaw as v, onBeforeUnmount as y, openBlock as h, createElementBlock as f, watch as M } from "vue";
import { ProjectModel as T } from "@bryntum/scheduler";
import { WrapperHelper as i } from "../helper/WrapperHelper.js";
const _ = /* @__PURE__ */ C({
  __name: "BryntumSchedulerProjectModel",
  props: {
    adjustDurationToDST: {},
    assignmentModelClass: {},
    assignments: {},
    assignmentsData: {},
    assignmentStore: {},
    assignmentStoreClass: {},
    children: {},
    dependencies: {},
    dependenciesData: {},
    dependencyModelClass: {},
    dependencyStore: {},
    dependencyStoreClass: {},
    eventModelClass: {},
    events: {},
    eventsData: {},
    eventStore: {},
    eventStoreClass: {},
    expanded: {},
    id: {},
    json: {},
    orderedParentIndex: {},
    parentId: {},
    parentIndex: {},
    readOnly: {},
    remoteChildCount: {},
    resourceModelClass: {},
    resources: {},
    resourcesData: {},
    resourceStore: {},
    resourceStoreClass: {},
    resourceTimeRanges: {},
    resourceTimeRangesData: {},
    resourceTimeRangeStore: {},
    resourceTimeRangeStoreClass: {},
    shouldSyncDataOnLoad: { type: Function },
    silenceInitialCommit: {},
    stm: {},
    timeRanges: {},
    timeRangesData: {},
    timeRangeStore: {},
    timeRangeStoreClass: {},
    timeZone: {},
    toJSONResultFormat: {},
    useRawData: {}
  },
  emits: ["change", "dataReady"],
  setup(d, { expose: l, emit: c }) {
    const m = d, u = c, a = {
      instanceClass: T,
      instanceName: "ProjectModel",
      configNames: [
        "adjustDurationToDST",
        "assignmentModelClass",
        "assignmentsData",
        "assignmentStoreClass",
        "children",
        "dependenciesData",
        "dependencyModelClass",
        "dependencyStoreClass",
        "eventModelClass",
        "eventsData",
        "eventStoreClass",
        "expanded",
        "orderedParentIndex",
        "parentId",
        "parentIndex",
        "resourceModelClass",
        "resourcesData",
        "resourceStoreClass",
        "resourceTimeRangesData",
        "resourceTimeRangeStoreClass",
        "silenceInitialCommit",
        "timeRangesData",
        "timeRangeStoreClass",
        "toJSONResultFormat",
        "useRawData"
      ],
      propertyConfigNames: [
        "assignments",
        "assignmentStore",
        "dependencies",
        "dependencyStore",
        "events",
        "eventStore",
        "id",
        "json",
        "onChange",
        "onDataReady",
        "readOnly",
        "remoteChildCount",
        "resources",
        "resourceStore",
        "resourceTimeRanges",
        "resourceTimeRangeStore",
        "shouldSyncDataOnLoad",
        "stm",
        "timeRanges",
        "timeRangeStore",
        "timeZone"
      ],
      propertyNames: [
        "allChildren",
        "descendantCount",
        "hasGeneratedId",
        "inlineData",
        "internalId",
        "isCommitting",
        "isCreating",
        "isValid",
        "previousSiblingsTotalCount",
        "visibleDescendantCount"
      ],
      eventNames: [
        "change",
        "dataReady"
      ]
    }, n = {}, s = r(), o = r(/* @__PURE__ */ new Map());
    return S("teleports", o), D(() => {
      const e = R();
      n.value = i.createWidget({
        me: e,
        props: m,
        emit: u,
        widgetData: a,
        element: s.value,
        toRaw: v
      });
      const t = (p, g) => M(() => e.props[p], g);
      i.watchProps(n.value, a, t);
    }), y(() => {
      var e, t;
      (t = (e = n.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), l({
      instance: n,
      refElement: s,
      teleports: o
    }), (e, t) => (h(), f("div", {
      ref_key: "refElement",
      ref: s
    }, null, 512));
  }
});
export {
  _
};
