// @bryntum/schedulerpro-vue-3/lib/chunks/BI7SwuHV.js
import { defineComponent as C, ref as a, provide as m, onMounted as h, getCurrentInstance as w, toRaw as b, onBeforeUnmount as R, openBlock as D, createElementBlock as S, watch as E } from "vue";
import { VersionGrid as M } from "@bryntum/schedulerpro";
import { WrapperHelper as i } from "../helper/WrapperHelper.js";
const v = /* @__PURE__ */ C({
  __name: "BryntumVersionGrid",
  props: {
    adopt: {},
    alignSelf: {},
    animateRemovingRows: {},
    animateTreeNodeToggle: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    autoHeight: {},
    bbar: {},
    bodyCls: {},
    bubbleEvents: {},
    callOnFunctions: {},
    cls: {},
    collapsed: {},
    collapsible: {},
    columnLines: {},
    columns: {},
    comparingVersionId: {},
    config: {},
    contentElementCls: {},
    contextMenuTriggerEvent: {},
    data: {},
    dataset: {},
    dateFormat: {},
    defaultRegion: {},
    destroyStore: {},
    disabled: {},
    disableGridColumnIdWarning: {},
    disableGridRowModelWarning: {},
    dock: {},
    emptyText: {},
    enableSticky: {},
    enableTextSelection: {},
    enableUndoRedoKeys: {},
    extraData: {},
    fillLastColumn: {},
    fixedRowHeight: {},
    flex: {},
    footer: {},
    fullRowRefresh: {},
    getRowHeight: { type: Function },
    header: {},
    height: {},
    hidden: {},
    hideFooters: {},
    hideHeaders: {},
    hideHorizontalScrollbar: {},
    hoverCls: {},
    id: {},
    ignoreParentReadOnly: {},
    insertBefore: {},
    insertFirst: {},
    keyMap: {},
    listeners: {},
    loadMask: {},
    loadMaskDefaults: {},
    loadMaskError: {},
    localizable: {},
    longPressTime: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxHeight: {},
    maxWidth: {},
    minHeight: {},
    minWidth: {},
    monitorResize: {},
    owner: {},
    plugins: {},
    preserveFocusOnDatasetChange: {},
    preserveScroll: {},
    preserveScrollOnDatasetChange: {},
    preventTooltipOnTouch: {},
    project: {},
    readOnly: {},
    relayStoreEvents: {},
    resizeToFitIncludesHeader: {},
    responsiveLevels: {},
    ripple: {},
    rootElement: {},
    rowHeight: {},
    rowLines: {},
    scrollable: {},
    scrollerClass: {},
    scrollManager: {},
    selectionMode: {},
    showDirty: {},
    showNamedVersionsOnly: {},
    showUnattachedTransactions: {},
    showVersions: {},
    stateful: {},
    statefulEvents: {},
    stateId: {},
    stateProvider: {},
    store: {},
    strips: {},
    subGridConfigs: {},
    syncMask: {},
    tab: {},
    tbar: {},
    title: {},
    tools: {},
    transitionDuration: {},
    type: {},
    ui: {},
    weight: {},
    width: {},
    cellCopyPasteFeature: {},
    cellEditFeature: {},
    cellMenuFeature: {},
    cellTooltipFeature: {},
    columnAutoWidthFeature: {},
    columnDragToolbarFeature: {},
    columnPickerFeature: {},
    columnRenameFeature: {},
    columnReorderFeature: {},
    columnResizeFeature: {},
    excelExporterFeature: {},
    fileDropFeature: {},
    fillHandleFeature: {},
    filterFeature: {},
    filterBarFeature: {},
    groupFeature: {},
    groupSummaryFeature: {},
    headerMenuFeature: {},
    lockRowsFeature: {},
    mergeCellsFeature: {},
    pdfExportFeature: {},
    printFeature: {},
    quickFindFeature: {},
    regionResizeFeature: {},
    rowCopyPasteFeature: {},
    rowExpanderFeature: {},
    rowReorderFeature: {},
    rowResizeFeature: {},
    searchFeature: {},
    sortFeature: {},
    splitFeature: {},
    stickyCellsFeature: {},
    stripeFeature: {},
    summaryFeature: {},
    treeFeature: {},
    treeGroupFeature: {}
  },
  emits: ["beforeCancelCellEdit", "beforeCellDelete", "beforeCellEditStart", "beforeColumnDragStart", "beforeColumnDropFinalize", "beforeColumnResize", "beforeCopy", "beforeCSVExport", "beforeDestroy", "beforeExcelExport", "beforeFillHandleDragStart", "beforeFinishCellEdit", "beforeHide", "beforePaste", "beforePdfExport", "beforeRenderRow", "beforeRenderRows", "beforeRowCollapse", "beforeRowExpand", "beforeSelectionChange", "beforeSetRecord", "beforeShow", "beforeToggleGroup", "beforeToggleNode", "cancelCellEdit", "catchAll", "cellClick", "cellContextMenu", "cellDblClick", "cellMenuBeforeShow", "cellMenuItem", "cellMenuShow", "cellMenuToggleItem", "cellMouseEnter", "cellMouseLeave", "cellMouseOut", "cellMouseOver", "collapse", "collapseNode", "columnDrag", "columnDragStart", "columnDrop", "columnResize", "columnResizeStart", "compare", "contextMenuItem", "contextMenuToggleItem", "copy", "dataChange", "destroy", "dragSelecting", "expand", "expandNode", "fileDrop", "fillHandleBeforeDragFinalize", "fillHandleDrag", "fillHandleDragAbort", "fillHandleDragEnd", "fillHandleDragStart", "finishCellEdit", "focusIn", "focusOut", "gridRowBeforeDragStart", "gridRowBeforeDropFinalize", "gridRowDrag", "gridRowDragAbort", "gridRowDragStart", "gridRowDrop", "headerClick", "headerMenuBeforeShow", "headerMenuItem", "headerMenuShow", "headerMenuToggleItem", "hide", "lockRows", "mouseOut", "mouseOver", "paint", "paste", "pdfExport", "recompose", "renderRow", "renderRows", "resize", "responsive", "restore", "rowCollapse", "rowExpand", "rowMouseEnter", "rowMouseLeave", "scroll", "selectionChange", "selectionModeChange", "show", "split", "splitterCollapseClick", "splitterDragEnd", "splitterDragStart", "splitterExpandClick", "startCellEdit", "stopCompare", "subGridCollapse", "subGridExpand", "toggleGroup", "toggleNode", "toolClick", "unlockRows", "unsplit"],
  setup(s, { expose: d, emit: c }) {
    const u = s, p = c, t = {
      instanceClass: M,
      instanceName: "VersionGrid",
      configNames: [
        "adopt",
        "animateRemovingRows",
        "ariaDescription",
        "ariaLabel",
        "autoHeight",
        "bbar",
        "bodyCls",
        "bubbleEvents",
        "callOnFunctions",
        "collapsible",
        "config",
        "contentElementCls",
        "contextMenuTriggerEvent",
        "dateFormat",
        "defaultRegion",
        "destroyStore",
        "disableGridColumnIdWarning",
        "disableGridRowModelWarning",
        "dock",
        "enableSticky",
        "enableTextSelection",
        "fillLastColumn",
        "fixedRowHeight",
        "footer",
        "fullRowRefresh",
        "getRowHeight",
        "header",
        "hideHorizontalScrollbar",
        "hoverCls",
        "ignoreParentReadOnly",
        "listeners",
        "loadMask",
        "loadMaskDefaults",
        "loadMaskError",
        "localizable",
        "maskDefaults",
        "masked",
        "monitorResize",
        "owner",
        "plugins",
        "preserveFocusOnDatasetChange",
        "preserveScrollOnDatasetChange",
        "preventTooltipOnTouch",
        "project",
        "relayStoreEvents",
        "resizeToFitIncludesHeader",
        "responsiveLevels",
        "ripple",
        "rootElement",
        "scrollerClass",
        "scrollManager",
        "showDirty",
        "stateful",
        "statefulEvents",
        "stateId",
        "stateProvider",
        "strips",
        "subGridConfigs",
        "syncMask",
        "tab",
        "tbar",
        "type",
        "ui",
        "weight"
      ],
      propertyConfigNames: [
        "alignSelf",
        "animateTreeNodeToggle",
        "appendTo",
        "ariaLive",
        "cls",
        "collapsed",
        "columnLines",
        "columns",
        "comparingVersionId",
        "data",
        "dataset",
        "disabled",
        "emptyText",
        "enableUndoRedoKeys",
        "extraData",
        "flex",
        "height",
        "hidden",
        "hideFooters",
        "hideHeaders",
        "id",
        "insertBefore",
        "insertFirst",
        "keyMap",
        "longPressTime",
        "margin",
        "maxHeight",
        "maxWidth",
        "minHeight",
        "minWidth",
        "onBeforeCancelCellEdit",
        "onBeforeCellDelete",
        "onBeforeCellEditStart",
        "onBeforeColumnDragStart",
        "onBeforeColumnDropFinalize",
        "onBeforeColumnResize",
        "onBeforeCopy",
        "onBeforeCSVExport",
        "onBeforeDestroy",
        "onBeforeExcelExport",
        "onBeforeFillHandleDragStart",
        "onBeforeFinishCellEdit",
        "onBeforeHide",
        "onBeforePaste",
        "onBeforePdfExport",
        "onBeforeRenderRow",
        "onBeforeRenderRows",
        "onBeforeRowCollapse",
        "onBeforeRowExpand",
        "onBeforeSelectionChange",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onBeforeToggleGroup",
        "onBeforeToggleNode",
        "onCancelCellEdit",
        "onCatchAll",
        "onCellClick",
        "onCellContextMenu",
        "onCellDblClick",
        "onCellMenuBeforeShow",
        "onCellMenuItem",
        "onCellMenuShow",
        "onCellMenuToggleItem",
        "onCellMouseEnter",
        "onCellMouseLeave",
        "onCellMouseOut",
        "onCellMouseOver",
        "onCollapse",
        "onCollapseNode",
        "onColumnDrag",
        "onColumnDragStart",
        "onColumnDrop",
        "onColumnResize",
        "onColumnResizeStart",
        "onCompare",
        "onContextMenuItem",
        "onContextMenuToggleItem",
        "onCopy",
        "onDataChange",
        "onDestroy",
        "onDragSelecting",
        "onExpand",
        "onExpandNode",
        "onFileDrop",
        "onFillHandleBeforeDragFinalize",
        "onFillHandleDrag",
        "onFillHandleDragAbort",
        "onFillHandleDragEnd",
        "onFillHandleDragStart",
        "onFinishCellEdit",
        "onFocusIn",
        "onFocusOut",
        "onGridRowBeforeDragStart",
        "onGridRowBeforeDropFinalize",
        "onGridRowDrag",
        "onGridRowDragAbort",
        "onGridRowDragStart",
        "onGridRowDrop",
        "onHeaderClick",
        "onHeaderMenuBeforeShow",
        "onHeaderMenuItem",
        "onHeaderMenuShow",
        "onHeaderMenuToggleItem",
        "onHide",
        "onLockRows",
        "onMouseOut",
        "onMouseOver",
        "onPaint",
        "onPaste",
        "onPdfExport",
        "onReadOnly",
        "onRecompose",
        "onRenderRow",
        "onRenderRows",
        "onResize",
        "onResponsive",
        "onRestore",
        "onRowCollapse",
        "onRowExpand",
        "onRowMouseEnter",
        "onRowMouseLeave",
        "onScroll",
        "onSelectionChange",
        "onSelectionModeChange",
        "onShow",
        "onSplit",
        "onSplitterCollapseClick",
        "onSplitterDragEnd",
        "onSplitterDragStart",
        "onSplitterExpandClick",
        "onStartCellEdit",
        "onStopCompare",
        "onSubGridCollapse",
        "onSubGridExpand",
        "onToggleGroup",
        "onToggleNode",
        "onToolClick",
        "onUnlockRows",
        "onUnsplit",
        "preserveScroll",
        "readOnly",
        "rowHeight",
        "rowLines",
        "scrollable",
        "selectionMode",
        "showNamedVersionsOnly",
        "showUnattachedTransactions",
        "showVersions",
        "store",
        "title",
        "tools",
        "transitionDuration",
        "width"
      ],
      propertyNames: [
        "accessibility",
        "originalStore",
        "selectedCell",
        "selectedCells",
        "selectedRecord",
        "selectedRecords",
        "selectedRows",
        "state",
        "tooltip"
      ],
      eventNames: [
        "beforeCancelCellEdit",
        "beforeCellDelete",
        "beforeCellEditStart",
        "beforeColumnDragStart",
        "beforeColumnDropFinalize",
        "beforeColumnResize",
        "beforeCopy",
        "beforeCSVExport",
        "beforeDestroy",
        "beforeExcelExport",
        "beforeFillHandleDragStart",
        "beforeFinishCellEdit",
        "beforeHide",
        "beforePaste",
        "beforePdfExport",
        "beforeRenderRow",
        "beforeRenderRows",
        "beforeRowCollapse",
        "beforeRowExpand",
        "beforeSelectionChange",
        "beforeSetRecord",
        "beforeShow",
        "beforeToggleGroup",
        "beforeToggleNode",
        "cancelCellEdit",
        "catchAll",
        "cellClick",
        "cellContextMenu",
        "cellDblClick",
        "cellMenuBeforeShow",
        "cellMenuItem",
        "cellMenuShow",
        "cellMenuToggleItem",
        "cellMouseEnter",
        "cellMouseLeave",
        "cellMouseOut",
        "cellMouseOver",
        "collapse",
        "collapseNode",
        "columnDrag",
        "columnDragStart",
        "columnDrop",
        "columnResize",
        "columnResizeStart",
        "compare",
        "contextMenuItem",
        "contextMenuToggleItem",
        "copy",
        "dataChange",
        "destroy",
        "dragSelecting",
        "expand",
        "expandNode",
        "fileDrop",
        "fillHandleBeforeDragFinalize",
        "fillHandleDrag",
        "fillHandleDragAbort",
        "fillHandleDragEnd",
        "fillHandleDragStart",
        "finishCellEdit",
        "focusIn",
        "focusOut",
        "gridRowBeforeDragStart",
        "gridRowBeforeDropFinalize",
        "gridRowDrag",
        "gridRowDragAbort",
        "gridRowDragStart",
        "gridRowDrop",
        "headerClick",
        "headerMenuBeforeShow",
        "headerMenuItem",
        "headerMenuShow",
        "headerMenuToggleItem",
        "hide",
        "lockRows",
        "mouseOut",
        "mouseOver",
        "paint",
        "paste",
        "pdfExport",
        "recompose",
        "renderRow",
        "renderRows",
        "resize",
        "responsive",
        "restore",
        "rowCollapse",
        "rowExpand",
        "rowMouseEnter",
        "rowMouseLeave",
        "scroll",
        "selectionChange",
        "selectionModeChange",
        "show",
        "split",
        "splitterCollapseClick",
        "splitterDragEnd",
        "splitterDragStart",
        "splitterExpandClick",
        "startCellEdit",
        "stopCompare",
        "subGridCollapse",
        "subGridExpand",
        "toggleGroup",
        "toggleNode",
        "toolClick",
        "unlockRows",
        "unsplit"
      ]
    }, r = {}, l = a(), n = a(/* @__PURE__ */ new Map());
    return m("teleports", n), h(() => {
      const e = w();
      r.value = i.createWidget({
        me: e,
        props: u,
        emit: p,
        widgetData: t,
        element: l.value,
        toRaw: b
      });
      const o = (g, f) => E(() => e.props[g], f);
      i.watchProps(r.value, t, o);
    }), R(() => {
      var e, o;
      (o = (e = r.value) == null ? void 0 : e.destroy) == null || o.call(e);
    }), d({
      instance: r,
      refElement: l,
      teleports: n
    }), (e, o) => (D(), S("div", {
      ref_key: "refElement",
      ref: l
    }, null, 512));
  }
});
export {
  v as _
};
