// @bryntum/schedulerpro-vue-3/lib/chunks/CB2kh9Rf.js
import { defineComponent as h, ref as a, provide as f, onMounted as y, getCurrentInstance as b, toRaw as S, onBeforeUnmount as w, openBlock as C, createElementBlock as T, watch as v } from "vue";
import { GroupBar as O } from "@bryntum/schedulerpro";
import { WrapperHelper as r } from "../helper/WrapperHelper.js";
const I = /* @__PURE__ */ h({
  __name: "BryntumGroupBar",
  props: {
    activateOnMouseover: {},
    adopt: {},
    align: {},
    alignSelf: {},
    allowGroupSelect: {},
    anchor: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    bubbleEvents: {},
    callOnFunctions: {},
    centered: {},
    clearSelectionOnEmptySpaceClick: {},
    closable: {},
    closeHandler: { type: [String, Function] },
    cls: {},
    collapsibleGroups: {},
    config: {},
    constrainTo: {},
    content: {},
    contentElementCls: {},
    dataset: {},
    defaultBindProperty: {},
    disabled: {},
    displayField: {},
    dock: {},
    draggable: {},
    extraData: {},
    flex: {},
    floating: {},
    getItemCls: { type: [Function, String] },
    getItemStyle: { type: [Function, String] },
    groupHeaderTpl: { type: Function },
    height: {},
    hidden: {},
    hideAnimation: {},
    html: { type: [String, Function, Array] },
    htmlCls: {},
    iconTpl: { type: Function },
    id: {},
    ignoreParentReadOnly: {},
    insertBefore: {},
    insertFirst: {},
    isSelectable: { type: [Function, String] },
    items: {},
    itemTpl: { type: Function },
    keyMap: {},
    listeners: {},
    localeClass: {},
    localizable: {},
    localizableProperties: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxHeight: {},
    maximizeOnMobile: {},
    maxWidth: {},
    minHeight: {},
    minWidth: {},
    monitorResize: {},
    multiSelect: {},
    owner: {},
    positioned: {},
    preventTooltipOnTouch: {},
    readOnly: {},
    relayStoreEvents: {},
    ripple: {},
    rootElement: {},
    scrollable: {},
    scrollAction: {},
    selectAllItem: {},
    selected: {},
    showAnimation: {},
    showTooltipWhenDisabled: {},
    store: {},
    tab: {},
    tag: {},
    textAlign: {},
    title: {},
    toggleAllIfCtrlPressed: {},
    tooltip: {},
    tooltipTemplate: { type: Function },
    type: {},
    ui: {},
    weight: {},
    width: {},
    x: {},
    y: {}
  },
  emits: ["beforeDestroy", "beforeHide", "beforeItem", "beforeShow", "catchAll", "destroy", "focusIn", "focusOut", "hide", "item", "paint", "recompose", "resize", "selectionChange", "show", "toggleGroup", "toggleNode"],
  setup(s, { expose: c, emit: p }) {
    const m = s, d = p, l = {
      instanceClass: O,
      instanceName: "GroupBar",
      configNames: [
        "activateOnMouseover",
        "adopt",
        "align",
        "allowGroupSelect",
        "anchor",
        "ariaDescription",
        "ariaLabel",
        "bubbleEvents",
        "callOnFunctions",
        "centered",
        "clearSelectionOnEmptySpaceClick",
        "closable",
        "closeHandler",
        "config",
        "constrainTo",
        "contentElementCls",
        "defaultBindProperty",
        "displayField",
        "dock",
        "draggable",
        "floating",
        "getItemCls",
        "getItemStyle",
        "groupHeaderTpl",
        "hideAnimation",
        "htmlCls",
        "iconTpl",
        "ignoreParentReadOnly",
        "isSelectable",
        "itemTpl",
        "listeners",
        "localeClass",
        "localizable",
        "localizableProperties",
        "maskDefaults",
        "masked",
        "monitorResize",
        "owner",
        "positioned",
        "preventTooltipOnTouch",
        "relayStoreEvents",
        "ripple",
        "rootElement",
        "scrollAction",
        "selectAllItem",
        "showAnimation",
        "showTooltipWhenDisabled",
        "tab",
        "tag",
        "textAlign",
        "tooltipTemplate",
        "type",
        "ui",
        "weight"
      ],
      propertyConfigNames: [
        "alignSelf",
        "appendTo",
        "ariaLive",
        "cls",
        "collapsibleGroups",
        "content",
        "dataset",
        "disabled",
        "extraData",
        "flex",
        "height",
        "hidden",
        "html",
        "id",
        "insertBefore",
        "insertFirst",
        "items",
        "keyMap",
        "margin",
        "maxHeight",
        "maximizeOnMobile",
        "maxWidth",
        "minHeight",
        "minWidth",
        "multiSelect",
        "onBeforeDestroy",
        "onBeforeHide",
        "onBeforeItem",
        "onBeforeShow",
        "onCatchAll",
        "onDestroy",
        "onFocusIn",
        "onFocusOut",
        "onHide",
        "onItem",
        "onPaint",
        "onReadOnly",
        "onRecompose",
        "onResize",
        "onSelectionChange",
        "onShow",
        "onToggleGroup",
        "onToggleNode",
        "readOnly",
        "scrollable",
        "selected",
        "store",
        "title",
        "toggleAllIfCtrlPressed",
        "tooltip",
        "width",
        "x",
        "y"
      ],
      propertyNames: [
        "accessibility",
        "anchorSize"
      ],
      eventNames: [
        "beforeDestroy",
        "beforeHide",
        "beforeItem",
        "beforeShow",
        "catchAll",
        "destroy",
        "focusIn",
        "focusOut",
        "hide",
        "item",
        "paint",
        "recompose",
        "resize",
        "selectionChange",
        "show",
        "toggleGroup",
        "toggleNode"
      ]
    }, o = {}, n = a(), i = a(/* @__PURE__ */ new Map());
    return f("teleports", i), y(() => {
      const e = b();
      o.value = r.createWidget({
        me: e,
        props: m,
        emit: d,
        widgetData: l,
        element: n.value,
        toRaw: S
      });
      const t = (g, u) => v(() => e.props[g], u);
      r.watchProps(o.value, l, t);
    }), w(() => {
      var e, t;
      (t = (e = o.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), c({
      instance: o,
      refElement: n,
      teleports: i
    }), (e, t) => (C(), T("div", {
      ref_key: "refElement",
      ref: n
    }, null, 512));
  }
});
export {
  I as _
};
