// @bryntum/schedulerpro-vue-3/lib/chunks/CXIOoWJg.js
import { defineComponent as S, ref as u, provide as D, onMounted as h, getCurrentInstance as E, toRaw as R, onBeforeUnmount as b, openBlock as a, createElementBlock as c, createElementVNode as F, Fragment as M, renderList as w, createBlock as d, Teleport as T, resolveDynamicComponent as k, mergeProps as y, watch as x } from "vue";
import { SchedulerProBase as B } from "@bryntum/schedulerpro";
import { WrapperHelper as r } from "../helper/WrapperHelper.js";
const z = {
  ref: "teleportsHolder",
  style: { display: "none" }
}, P = /* @__PURE__ */ S({
  __name: "BryntumSchedulerProBase",
  props: {
    adopt: {},
    alignSelf: {},
    allowCreate: {},
    allowDropOnEventBar: {},
    allowOverlap: {},
    animateRemovingRows: {},
    animateTreeNodeToggle: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    assignments: {},
    assignmentStore: {},
    autoAdjustTimeAxis: {},
    autoHeight: {},
    barMargin: {},
    bbar: {},
    bodyCls: {},
    bubbleEvents: {},
    bufferCoef: {},
    bufferThreshold: {},
    calendars: {},
    callOnFunctions: {},
    cls: {},
    collapsed: {},
    collapsible: {},
    columnLines: {},
    columns: {},
    config: {},
    contentElementCls: {},
    contextMenuTriggerEvent: {},
    createEventOnDblClick: {},
    creationTooltip: {},
    creationTooltipTemplate: { type: Function },
    crudManager: {},
    crudManagerClass: {},
    cycleResolutionPopupClass: {},
    data: {},
    dataset: {},
    date: {},
    dateFormat: {},
    dateSeparator: {},
    defaultRegion: {},
    defaultResourceImageName: {},
    dependencies: {},
    dependencyIdField: {},
    dependencyStore: {},
    descriptionFormat: {},
    descriptionRenderer: { type: Function },
    deselectAllOnScheduleClick: {},
    deselectOnClick: {},
    destroyStore: {},
    destroyStores: {},
    disabled: {},
    disableGridColumnIdWarning: {},
    disableGridRowModelWarning: {},
    displayDateFormat: {},
    displaySchedulingIssueResolutionPopup: {},
    dock: {},
    durationDisplayPrecision: {},
    emptyText: {},
    enableDeleteKey: {},
    enableEventAnimations: {},
    enableRecurringEvents: {},
    enableSticky: {},
    enableTextSelection: {},
    enableTransactionalFeatures: {},
    enableUndoRedoKeys: {},
    endDate: {},
    endParamName: {},
    eventBarTextField: {},
    eventColor: {},
    eventLayout: {},
    eventRenderer: { type: Function },
    eventRendererThisObj: {},
    events: {},
    eventSelectionDisabled: {},
    eventStore: {},
    eventStyle: {},
    extraData: {},
    fillLastColumn: {},
    fillTicks: {},
    fixedRowHeight: {},
    flex: {},
    footer: {},
    forceFit: {},
    fullRowRefresh: {},
    getDateConstraints: { type: Function },
    getRowHeight: { type: Function },
    header: {},
    height: {},
    hidden: {},
    hideFooters: {},
    hideHeaders: {},
    hideHorizontalScrollbar: {},
    hideRowHover: {},
    highlightPredecessors: {},
    highlightSuccessors: {},
    hoverCls: {},
    id: {},
    ignoreDomEventsWhileScrolling: {},
    ignoreParentReadOnly: {},
    infiniteScroll: {},
    insertBefore: {},
    insertFirst: {},
    isEventSelectable: { type: Function },
    keyMap: {},
    listeners: {},
    loadMask: {},
    loadMaskDefaults: {},
    loadMaskError: {},
    localizable: {},
    longPressTime: {},
    maintainSelectionOnDatasetChange: {},
    managedEventSizing: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxHeight: {},
    maxTimeAxisUnit: {},
    maxWidth: {},
    maxZoomLevel: {},
    milestoneAlign: {},
    milestoneCharWidth: {},
    milestoneLayoutMode: {},
    milestoneTextPosition: {},
    minHeight: {},
    minWidth: {},
    minZoomLevel: {},
    mode: {},
    monitorResize: {},
    multiEventSelect: {},
    narrowEventWidth: {},
    overlappingEventSorter: { type: Function },
    owner: {},
    partner: {},
    passStartEndParameters: {},
    plugins: {},
    preCalculateHeightLimit: {},
    preserveFocusOnDatasetChange: {},
    preserveScroll: {},
    preserveScrollOnDatasetChange: {},
    presets: {},
    preventTooltipOnTouch: {},
    project: {},
    projectProgressReporting: {},
    range: {},
    readOnly: {},
    relayStoreEvents: {},
    resizeToFitIncludesHeader: {},
    resourceColumns: {},
    resourceImageExtension: {},
    resourceImagePath: {},
    resourceMargin: {},
    resources: {},
    resourceStore: {},
    resourceTimeRangeRenderer: { type: Function },
    resourceTimeRanges: {},
    resourceTimeRangeStore: {},
    responsiveLevels: {},
    ripple: {},
    rootElement: {},
    rowHeight: {},
    rowLines: {},
    schedulingIssueResolutionPopupClass: {},
    scrollable: {},
    scrollerClass: {},
    scrollExtensionThreshold: {},
    scrollManager: {},
    selectionMode: {},
    selectResourceOnEventNavigate: {},
    selectResourceOnScheduleClick: {},
    showCreationTooltip: {},
    showDirty: {},
    showEventColorPickers: {},
    showLagInTooltip: {},
    showRecurringUI: {},
    showTaskColorPickers: {},
    showTooltip: {},
    snap: {},
    snapRelativeToEventStartDate: {},
    startDate: {},
    startParamName: {},
    stateful: {},
    statefulEvents: {},
    stateId: {},
    stateProvider: {},
    stepUnit: {},
    stickyHeaders: {},
    store: {},
    strips: {},
    subGridConfigs: {},
    suppressFit: {},
    syncMask: {},
    tab: {},
    tbar: {},
    terminalCls: {},
    terminalHideDelay: {},
    terminalOffset: {},
    terminalShowDelay: {},
    terminalSides: {},
    terminalSize: {},
    tickSize: {},
    timeAxis: {},
    timeRanges: {},
    timeRangeStore: {},
    timeZone: {},
    title: {},
    tools: {},
    tooltipTemplate: { type: Function },
    transitionDuration: {},
    triggerSelectionChangeOnRemove: {},
    ui: {},
    updateTimelineContextOnScroll: {},
    useContextualRecurrenceRules: {},
    useInitialAnimation: {},
    verticalTimeAxisColumn: {},
    viewPreset: {},
    visibleDate: {},
    visibleZoomFactor: {},
    weekStartDay: {},
    weight: {},
    width: {},
    workingTime: {},
    zoomKeepsOriginalTimespan: {},
    zoomOnMouseWheel: {},
    zoomOnTimeAxisDoubleClick: {},
    calendarHighlightFeature: {},
    cellCopyPasteFeature: {},
    cellEditFeature: {},
    cellMenuFeature: {},
    cellTooltipFeature: {},
    columnAutoWidthFeature: {},
    columnDragToolbarFeature: {},
    columnLinesFeature: {},
    columnPickerFeature: {},
    columnRenameFeature: {},
    columnReorderFeature: {},
    columnResizeFeature: {},
    dependenciesFeature: {},
    dependencyEditFeature: {},
    eventBufferFeature: {},
    eventCopyPasteFeature: {},
    eventDragFeature: {},
    eventDragCreateFeature: {},
    eventDragSelectFeature: {},
    eventEditFeature: {},
    eventFilterFeature: {},
    eventMenuFeature: {},
    eventNonWorkingTimeFeature: {},
    eventResizeFeature: {},
    eventSegmentDragFeature: {},
    eventSegmentResizeFeature: {},
    eventSegmentsFeature: {},
    eventTooltipFeature: {},
    excelExporterFeature: {},
    fileDropFeature: {},
    fillHandleFeature: {},
    filterFeature: {},
    filterBarFeature: {},
    groupFeature: {},
    groupSummaryFeature: {},
    headerMenuFeature: {},
    headerZoomFeature: {},
    labelsFeature: {},
    lockRowsFeature: {},
    mergeCellsFeature: {},
    nestedEventsFeature: {},
    nonWorkingTimeFeature: {},
    panFeature: {},
    pdfExportFeature: {},
    percentBarFeature: {},
    printFeature: {},
    quickFindFeature: {},
    regionResizeFeature: {},
    resourceMenuFeature: {},
    resourceNonWorkingTimeFeature: {},
    resourceTimeRangesFeature: {},
    rowCopyPasteFeature: {},
    rowExpanderFeature: {},
    rowReorderFeature: {},
    rowResizeFeature: {},
    scheduleContextFeature: {},
    scheduleMenuFeature: {},
    scheduleTooltipFeature: {},
    scrollButtonsFeature: {},
    searchFeature: {},
    simpleEventEditFeature: {},
    sortFeature: {},
    splitFeature: {},
    stickyCellsFeature: {},
    stickyEventsFeature: {},
    stripeFeature: {},
    summaryFeature: {},
    taskEditFeature: {},
    timeAxisHeaderMenuFeature: {},
    timeRangesFeature: {},
    timeSelectionFeature: {},
    timeSpanHighlightFeature: {},
    treeFeature: {},
    treeGroupFeature: {},
    versionsFeature: {}
  },
  emits: ["afterDependencyCreateDrop", "afterDependencySave", "afterDragCreate", "afterEventDrop", "afterEventEdit", "afterEventSave", "afterEventSegmentDrop", "afterTaskEdit", "afterTaskSave", "assignmentSelectionChange", "beforeAssignmentDelete", "beforeAssignmentSelectionChange", "beforeCancelCellEdit", "beforeCellDelete", "beforeCellEditStart", "beforeColumnDragStart", "beforeColumnDropFinalize", "beforeColumnResize", "beforeCopy", "beforeCSVExport", "beforeDependencyAdd", "beforeDependencyCreateDrag", "beforeDependencyCreateFinalize", "beforeDependencyDelete", "beforeDependencyEdit", "beforeDependencyEditShow", "beforeDependencySave", "beforeDestroy", "beforeDragCreate", "beforeDragCreateFinalize", "beforeEventAdd", "beforeEventDelete", "beforeEventDrag", "beforeEventDragSelect", "beforeEventDropFinalize", "beforeEventEdit", "beforeEventEditShow", "beforeEventResize", "beforeEventResizeFinalize", "beforeEventSave", "beforeEventSegmentDrag", "beforeEventSegmentDropFinalize", "beforeEventSegmentResize", "beforeEventSegmentResizeFinalize", "beforeEventSelectionChange", "beforeExcelExport", "beforeFillHandleDragStart", "beforeFinishCellEdit", "beforeHide", "beforePan", "beforePaste", "beforePdfExport", "beforePresetChange", "beforeRenderRow", "beforeRenderRows", "beforeRowCollapse", "beforeRowExpand", "beforeSelectionChange", "beforeSetRecord", "beforeShow", "beforeShowTerminals", "beforeTaskDelete", "beforeTaskEdit", "beforeTaskEditShow", "beforeTaskSave", "beforeToggleGroup", "beforeToggleNode", "cancelCellEdit", "catchAll", "cellClick", "cellContextMenu", "cellDblClick", "cellMenuBeforeShow", "cellMenuItem", "cellMenuShow", "cellMenuToggleItem", "cellMouseEnter", "cellMouseLeave", "cellMouseOut", "cellMouseOver", "collapse", "collapseNode", "columnDrag", "columnDragStart", "columnDrop", "columnResize", "columnResizeStart", "contextMenuItem", "contextMenuToggleItem", "copy", "currentTimelineUpdate", "dataChange", "dateRangeChange", "dependenciesDrawn", "dependencyClick", "dependencyContextMenu", "dependencyCreateDragStart", "dependencyCreateDrop", "dependencyDblClick", "dependencyMouseOut", "dependencyMouseOver", "dependencyValidationComplete", "dependencyValidationStart", "destroy", "dragCreateEnd", "dragCreateStart", "dragSelecting", "eventAutoCreated", "eventClick", "eventContextMenu", "eventDblClick", "eventDrag", "eventDragAbort", "eventDragModeChange", "eventDragReset", "eventDragStart", "eventDrop", "eventEditBeforeSetRecord", "eventKeyDown", "eventKeyUp", "eventMenuBeforeShow", "eventMenuItem", "eventMenuShow", "eventMouseDown", "eventMouseEnter", "eventMouseLeave", "eventMouseOut", "eventMouseOver", "eventMouseUp", "eventPartialResize", "eventResizeEnd", "eventResizeStart", "eventSegmentDrag", "eventSegmentDragAbort", "eventSegmentDragReset", "eventSegmentDragStart", "eventSegmentDrop", "eventSegmentPartialResize", "eventSegmentResizeEnd", "eventSegmentResizeStart", "eventSelectionChange", "expand", "expandNode", "fileDrop", "fillHandleBeforeDragFinalize", "fillHandleDrag", "fillHandleDragAbort", "fillHandleDragEnd", "fillHandleDragStart", "finishCellEdit", "focusIn", "focusOut", "gridRowBeforeDragStart", "gridRowBeforeDropFinalize", "gridRowDrag", "gridRowDragAbort", "gridRowDragStart", "gridRowDrop", "headerClick", "headerMenuBeforeShow", "headerMenuItem", "headerMenuShow", "headerMenuToggleItem", "hide", "lockRows", "mouseOut", "mouseOver", "navigate", "paint", "paste", "pasteNotAllowed", "pdfExport", "percentBarDrag", "percentBarDragAbort", "percentBarDragStart", "percentBarDrop", "presetChange", "recompose", "releaseEvent", "renderEvent", "renderRow", "renderRows", "resize", "resourceHeaderClick", "resourceHeaderContextmenu", "resourceHeaderDblclick", "resourceMenuBeforeShow", "resourceMenuItem", "resourceMenuShow", "resourceMenuToggleItem", "resourceNonWorkingTimeClick", "resourceNonWorkingTimeContextMenu", "resourceNonWorkingTimeDblClick", "resourceNonWorkingTimeMouseDown", "resourceNonWorkingTimeMouseOut", "resourceNonWorkingTimeMouseOver", "resourceNonWorkingTimeMouseUp", "resourceTimeRangeClick", "resourceTimeRangeContextMenu", "resourceTimeRangeDblClick", "resourceTimeRangeMouseDown", "resourceTimeRangeMouseOut", "resourceTimeRangeMouseOver", "resourceTimeRangeMouseUp", "responsive", "rowCollapse", "rowExpand", "rowMouseEnter", "rowMouseLeave", "scheduleClick", "scheduleContextMenu", "scheduleDblClick", "scheduleMenuBeforeShow", "scheduleMenuItem", "scheduleMenuShow", "scheduleMouseDown", "scheduleMouseEnter", "scheduleMouseLeave", "scheduleMouseMove", "scheduleMouseUp", "scroll", "scrollButtonClick", "selectionChange", "selectionModeChange", "show", "split", "splitterCollapseClick", "splitterDragEnd", "splitterDragStart", "splitterExpandClick", "startCellEdit", "subGridCollapse", "subGridExpand", "taskEditCanceled", "timeAxisChange", "timeAxisHeaderClick", "timeAxisHeaderContextMenu", "timeAxisHeaderDblClick", "timeAxisHeaderMenuBeforeShow", "timeAxisHeaderMenuItem", "timeAxisHeaderMenuShow", "timelineContextChange", "timelineViewportResize", "timeRangeHeaderClick", "timeRangeHeaderContextMenu", "timeRangeHeaderDblClick", "timeSelectionChange", "timeSelectionElementClick", "toggleGroup", "toggleNode", "toolClick", "transactionChange", "unlockRows", "unsplit", "visibleDateRangeChange"],
  setup(g, { expose: m, emit: p }) {
    const v = g, C = p, s = {
      instanceClass: B,
      instanceName: "SchedulerProBase",
      isView: !0,
      dataStores: {
        assignmentStore: "assignments",
        calendarManagerStore: "calendars",
        dependencyStore: "dependencies",
        eventStore: "events",
        resourceStore: "resources",
        resourceTimeRangeStore: "resourceTimeRanges",
        timeRangeStore: "timeRanges"
      },
      projectStores: !0,
      featureNames: [
        "calendarHighlightFeature",
        "cellCopyPasteFeature",
        "cellEditFeature",
        "cellMenuFeature",
        "cellTooltipFeature",
        "columnAutoWidthFeature",
        "columnDragToolbarFeature",
        "columnLinesFeature",
        "columnPickerFeature",
        "columnRenameFeature",
        "columnReorderFeature",
        "columnResizeFeature",
        "dependenciesFeature",
        "dependencyEditFeature",
        "eventBufferFeature",
        "eventCopyPasteFeature",
        "eventDragFeature",
        "eventDragCreateFeature",
        "eventDragSelectFeature",
        "eventEditFeature",
        "eventFilterFeature",
        "eventMenuFeature",
        "eventNonWorkingTimeFeature",
        "eventResizeFeature",
        "eventSegmentDragFeature",
        "eventSegmentResizeFeature",
        "eventSegmentsFeature",
        "eventTooltipFeature",
        "excelExporterFeature",
        "fileDropFeature",
        "fillHandleFeature",
        "filterFeature",
        "filterBarFeature",
        "groupFeature",
        "groupSummaryFeature",
        "headerMenuFeature",
        "headerZoomFeature",
        "labelsFeature",
        "lockRowsFeature",
        "mergeCellsFeature",
        "nestedEventsFeature",
        "nonWorkingTimeFeature",
        "panFeature",
        "pdfExportFeature",
        "percentBarFeature",
        "printFeature",
        "quickFindFeature",
        "regionResizeFeature",
        "resourceMenuFeature",
        "resourceNonWorkingTimeFeature",
        "resourceTimeRangesFeature",
        "rowCopyPasteFeature",
        "rowExpanderFeature",
        "rowReorderFeature",
        "rowResizeFeature",
        "scheduleContextFeature",
        "scheduleMenuFeature",
        "scheduleTooltipFeature",
        "scrollButtonsFeature",
        "searchFeature",
        "simpleEventEditFeature",
        "sortFeature",
        "splitFeature",
        "stickyCellsFeature",
        "stickyEventsFeature",
        "stripeFeature",
        "summaryFeature",
        "taskEditFeature",
        "timeAxisHeaderMenuFeature",
        "timeRangesFeature",
        "timeSelectionFeature",
        "timeSpanHighlightFeature",
        "treeFeature",
        "treeGroupFeature",
        "versionsFeature"
      ],
      configNames: [
        "adopt",
        "allowCreate",
        "animateRemovingRows",
        "animateTreeNodeToggle",
        "ariaDescription",
        "ariaLabel",
        "autoAdjustTimeAxis",
        "autoHeight",
        "bbar",
        "bodyCls",
        "bubbleEvents",
        "bufferCoef",
        "bufferThreshold",
        "callOnFunctions",
        "collapsible",
        "config",
        "contentElementCls",
        "contextMenuTriggerEvent",
        "createEventOnDblClick",
        "creationTooltip",
        "crudManagerClass",
        "cycleResolutionPopupClass",
        "date",
        "defaultRegion",
        "defaultResourceImageName",
        "dependencyIdField",
        "descriptionRenderer",
        "deselectAllOnScheduleClick",
        "deselectOnClick",
        "destroyStore",
        "destroyStores",
        "disableGridColumnIdWarning",
        "disableGridRowModelWarning",
        "displaySchedulingIssueResolutionPopup",
        "dock",
        "durationDisplayPrecision",
        "enableDeleteKey",
        "enableRecurringEvents",
        "enableSticky",
        "enableTextSelection",
        "enableTransactionalFeatures",
        "endParamName",
        "eventBarTextField",
        "eventRenderer",
        "eventRendererThisObj",
        "eventSelectionDisabled",
        "fillLastColumn",
        "fixedRowHeight",
        "footer",
        "fullRowRefresh",
        "getDateConstraints",
        "getRowHeight",
        "header",
        "hideHorizontalScrollbar",
        "highlightPredecessors",
        "highlightSuccessors",
        "hoverCls",
        "ignoreDomEventsWhileScrolling",
        "ignoreParentReadOnly",
        "infiniteScroll",
        "listeners",
        "loadMask",
        "loadMaskDefaults",
        "loadMaskError",
        "localizable",
        "maintainSelectionOnDatasetChange",
        "managedEventSizing",
        "maskDefaults",
        "masked",
        "maxTimeAxisUnit",
        "mode",
        "monitorResize",
        "multiEventSelect",
        "narrowEventWidth",
        "owner",
        "partner",
        "passStartEndParameters",
        "plugins",
        "preCalculateHeightLimit",
        "preserveFocusOnDatasetChange",
        "preserveScrollOnDatasetChange",
        "preventTooltipOnTouch",
        "projectProgressReporting",
        "range",
        "relayStoreEvents",
        "resizeToFitIncludesHeader",
        "resourceColumns",
        "resourceImageExtension",
        "resourceImagePath",
        "resourceTimeRangeRenderer",
        "responsiveLevels",
        "ripple",
        "rootElement",
        "schedulingIssueResolutionPopupClass",
        "scrollerClass",
        "scrollExtensionThreshold",
        "scrollManager",
        "showCreationTooltip",
        "showDirty",
        "showEventColorPickers",
        "showLagInTooltip",
        "showRecurringUI",
        "showTaskColorPickers",
        "snapRelativeToEventStartDate",
        "startParamName",
        "stateful",
        "statefulEvents",
        "stateId",
        "stateProvider",
        "stepUnit",
        "stickyHeaders",
        "strips",
        "subGridConfigs",
        "suppressFit",
        "syncMask",
        "tab",
        "tbar",
        "terminalCls",
        "terminalSides",
        "timeAxis",
        "triggerSelectionChangeOnRemove",
        "ui",
        "verticalTimeAxisColumn",
        "visibleZoomFactor",
        "weekStartDay",
        "weight",
        "zoomKeepsOriginalTimespan",
        "zoomOnMouseWheel",
        "zoomOnTimeAxisDoubleClick"
      ],
      propertyConfigNames: [
        "alignSelf",
        "allowDropOnEventBar",
        "allowOverlap",
        "appendTo",
        "ariaLive",
        "assignments",
        "assignmentStore",
        "barMargin",
        "calendars",
        "cls",
        "collapsed",
        "columnLines",
        "columns",
        "creationTooltipTemplate",
        "crudManager",
        "data",
        "dataset",
        "dateFormat",
        "dateSeparator",
        "dependencies",
        "dependencyStore",
        "descriptionFormat",
        "disabled",
        "displayDateFormat",
        "emptyText",
        "enableEventAnimations",
        "enableUndoRedoKeys",
        "endDate",
        "eventColor",
        "eventLayout",
        "events",
        "eventStore",
        "eventStyle",
        "extraData",
        "fillTicks",
        "flex",
        "forceFit",
        "height",
        "hidden",
        "hideFooters",
        "hideHeaders",
        "hideRowHover",
        "id",
        "insertBefore",
        "insertFirst",
        "isEventSelectable",
        "keyMap",
        "longPressTime",
        "margin",
        "maxHeight",
        "maxWidth",
        "maxZoomLevel",
        "milestoneAlign",
        "milestoneCharWidth",
        "milestoneLayoutMode",
        "milestoneTextPosition",
        "minHeight",
        "minWidth",
        "minZoomLevel",
        "onAfterDependencyCreateDrop",
        "onAfterDependencySave",
        "onAfterDragCreate",
        "onAfterEventDrop",
        "onAfterEventEdit",
        "onAfterEventSave",
        "onAfterEventSegmentDrop",
        "onAfterTaskEdit",
        "onAfterTaskSave",
        "onAssignmentSelectionChange",
        "onBeforeAssignmentDelete",
        "onBeforeAssignmentSelectionChange",
        "onBeforeCancelCellEdit",
        "onBeforeCellDelete",
        "onBeforeCellEditStart",
        "onBeforeColumnDragStart",
        "onBeforeColumnDropFinalize",
        "onBeforeColumnResize",
        "onBeforeCopy",
        "onBeforeCSVExport",
        "onBeforeDependencyAdd",
        "onBeforeDependencyCreateDrag",
        "onBeforeDependencyCreateFinalize",
        "onBeforeDependencyDelete",
        "onBeforeDependencyEdit",
        "onBeforeDependencyEditShow",
        "onBeforeDependencySave",
        "onBeforeDestroy",
        "onBeforeDragCreate",
        "onBeforeDragCreateFinalize",
        "onBeforeEventAdd",
        "onBeforeEventDelete",
        "onBeforeEventDrag",
        "onBeforeEventDragSelect",
        "onBeforeEventDropFinalize",
        "onBeforeEventEdit",
        "onBeforeEventEditShow",
        "onBeforeEventResize",
        "onBeforeEventResizeFinalize",
        "onBeforeEventSave",
        "onBeforeEventSegmentDrag",
        "onBeforeEventSegmentDropFinalize",
        "onBeforeEventSegmentResize",
        "onBeforeEventSegmentResizeFinalize",
        "onBeforeEventSelectionChange",
        "onBeforeExcelExport",
        "onBeforeFillHandleDragStart",
        "onBeforeFinishCellEdit",
        "onBeforeHide",
        "onBeforePan",
        "onBeforePaste",
        "onBeforePdfExport",
        "onBeforePresetChange",
        "onBeforeRenderRow",
        "onBeforeRenderRows",
        "onBeforeRowCollapse",
        "onBeforeRowExpand",
        "onBeforeSelectionChange",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onBeforeShowTerminals",
        "onBeforeTaskDelete",
        "onBeforeTaskEdit",
        "onBeforeTaskEditShow",
        "onBeforeTaskSave",
        "onBeforeToggleGroup",
        "onBeforeToggleNode",
        "onCancelCellEdit",
        "onCatchAll",
        "onCellClick",
        "onCellContextMenu",
        "onCellDblClick",
        "onCellMenuBeforeShow",
        "onCellMenuItem",
        "onCellMenuShow",
        "onCellMenuToggleItem",
        "onCellMouseEnter",
        "onCellMouseLeave",
        "onCellMouseOut",
        "onCellMouseOver",
        "onCollapse",
        "onCollapseNode",
        "onColumnDrag",
        "onColumnDragStart",
        "onColumnDrop",
        "onColumnResize",
        "onColumnResizeStart",
        "onContextMenuItem",
        "onContextMenuToggleItem",
        "onCopy",
        "onCurrentTimelineUpdate",
        "onDataChange",
        "onDateRangeChange",
        "onDependenciesDrawn",
        "onDependencyClick",
        "onDependencyContextMenu",
        "onDependencyCreateDragStart",
        "onDependencyCreateDrop",
        "onDependencyDblClick",
        "onDependencyMouseOut",
        "onDependencyMouseOver",
        "onDependencyValidationComplete",
        "onDependencyValidationStart",
        "onDestroy",
        "onDragCreateEnd",
        "onDragCreateStart",
        "onDragSelecting",
        "onEventAutoCreated",
        "onEventClick",
        "onEventContextMenu",
        "onEventDblClick",
        "onEventDrag",
        "onEventDragAbort",
        "onEventDragModeChange",
        "onEventDragReset",
        "onEventDragStart",
        "onEventDrop",
        "onEventEditBeforeSetRecord",
        "onEventKeyDown",
        "onEventKeyUp",
        "onEventMenuBeforeShow",
        "onEventMenuItem",
        "onEventMenuShow",
        "onEventMouseDown",
        "onEventMouseEnter",
        "onEventMouseLeave",
        "onEventMouseOut",
        "onEventMouseOver",
        "onEventMouseUp",
        "onEventPartialResize",
        "onEventResizeEnd",
        "onEventResizeStart",
        "onEventSegmentDrag",
        "onEventSegmentDragAbort",
        "onEventSegmentDragReset",
        "onEventSegmentDragStart",
        "onEventSegmentDrop",
        "onEventSegmentPartialResize",
        "onEventSegmentResizeEnd",
        "onEventSegmentResizeStart",
        "onEventSelectionChange",
        "onExpand",
        "onExpandNode",
        "onFileDrop",
        "onFillHandleBeforeDragFinalize",
        "onFillHandleDrag",
        "onFillHandleDragAbort",
        "onFillHandleDragEnd",
        "onFillHandleDragStart",
        "onFinishCellEdit",
        "onFocusIn",
        "onFocusOut",
        "onGridRowBeforeDragStart",
        "onGridRowBeforeDropFinalize",
        "onGridRowDrag",
        "onGridRowDragAbort",
        "onGridRowDragStart",
        "onGridRowDrop",
        "onHeaderClick",
        "onHeaderMenuBeforeShow",
        "onHeaderMenuItem",
        "onHeaderMenuShow",
        "onHeaderMenuToggleItem",
        "onHide",
        "onLockRows",
        "onMouseOut",
        "onMouseOver",
        "onNavigate",
        "onPaint",
        "onPaste",
        "onPasteNotAllowed",
        "onPdfExport",
        "onPercentBarDrag",
        "onPercentBarDragAbort",
        "onPercentBarDragStart",
        "onPercentBarDrop",
        "onPresetChange",
        "onReadOnly",
        "onRecompose",
        "onReleaseEvent",
        "onRenderEvent",
        "onRenderRow",
        "onRenderRows",
        "onResize",
        "onResourceHeaderClick",
        "onResourceHeaderContextmenu",
        "onResourceHeaderDblclick",
        "onResourceMenuBeforeShow",
        "onResourceMenuItem",
        "onResourceMenuShow",
        "onResourceMenuToggleItem",
        "onResourceNonWorkingTimeClick",
        "onResourceNonWorkingTimeContextMenu",
        "onResourceNonWorkingTimeDblClick",
        "onResourceNonWorkingTimeMouseDown",
        "onResourceNonWorkingTimeMouseOut",
        "onResourceNonWorkingTimeMouseOver",
        "onResourceNonWorkingTimeMouseUp",
        "onResourceTimeRangeClick",
        "onResourceTimeRangeContextMenu",
        "onResourceTimeRangeDblClick",
        "onResourceTimeRangeMouseDown",
        "onResourceTimeRangeMouseOut",
        "onResourceTimeRangeMouseOver",
        "onResourceTimeRangeMouseUp",
        "onResponsive",
        "onRowCollapse",
        "onRowExpand",
        "onRowMouseEnter",
        "onRowMouseLeave",
        "onScheduleClick",
        "onScheduleContextMenu",
        "onScheduleDblClick",
        "onScheduleMenuBeforeShow",
        "onScheduleMenuItem",
        "onScheduleMenuShow",
        "onScheduleMouseDown",
        "onScheduleMouseEnter",
        "onScheduleMouseLeave",
        "onScheduleMouseMove",
        "onScheduleMouseUp",
        "onScroll",
        "onScrollButtonClick",
        "onSelectionChange",
        "onSelectionModeChange",
        "onShow",
        "onSplit",
        "onSplitterCollapseClick",
        "onSplitterDragEnd",
        "onSplitterDragStart",
        "onSplitterExpandClick",
        "onStartCellEdit",
        "onSubGridCollapse",
        "onSubGridExpand",
        "onTaskEditCanceled",
        "onTimeAxisChange",
        "onTimeAxisHeaderClick",
        "onTimeAxisHeaderContextMenu",
        "onTimeAxisHeaderDblClick",
        "onTimeAxisHeaderMenuBeforeShow",
        "onTimeAxisHeaderMenuItem",
        "onTimeAxisHeaderMenuShow",
        "onTimelineContextChange",
        "onTimelineViewportResize",
        "onTimeRangeHeaderClick",
        "onTimeRangeHeaderContextMenu",
        "onTimeRangeHeaderDblClick",
        "onTimeSelectionChange",
        "onTimeSelectionElementClick",
        "onToggleGroup",
        "onToggleNode",
        "onToolClick",
        "onTransactionChange",
        "onUnlockRows",
        "onUnsplit",
        "onVisibleDateRangeChange",
        "overlappingEventSorter",
        "preserveScroll",
        "presets",
        "project",
        "readOnly",
        "resourceMargin",
        "resources",
        "resourceStore",
        "resourceTimeRanges",
        "resourceTimeRangeStore",
        "rowHeight",
        "rowLines",
        "scrollable",
        "selectionMode",
        "selectResourceOnEventNavigate",
        "selectResourceOnScheduleClick",
        "showTooltip",
        "snap",
        "startDate",
        "store",
        "terminalHideDelay",
        "terminalOffset",
        "terminalShowDelay",
        "terminalSize",
        "tickSize",
        "timeRanges",
        "timeRangeStore",
        "timeZone",
        "title",
        "tools",
        "tooltipTemplate",
        "transitionDuration",
        "updateTimelineContextOnScroll",
        "useContextualRecurrenceRules",
        "useInitialAnimation",
        "viewPreset",
        "visibleDate",
        "width",
        "workingTime"
      ],
      propertyNames: [
        "accessibility",
        "isEngineReady",
        "originalStore",
        "scrollLeft",
        "scrollTop",
        "scrollX",
        "selectedAssignments",
        "selectedCell",
        "selectedCells",
        "selectedEvents",
        "selectedRecord",
        "selectedRecords",
        "selectedRows",
        "state",
        "timeResolution",
        "tooltip",
        "zoomLevel"
      ],
      eventNames: [
        "afterDependencyCreateDrop",
        "afterDependencySave",
        "afterDragCreate",
        "afterEventDrop",
        "afterEventEdit",
        "afterEventSave",
        "afterEventSegmentDrop",
        "afterTaskEdit",
        "afterTaskSave",
        "assignmentSelectionChange",
        "beforeAssignmentDelete",
        "beforeAssignmentSelectionChange",
        "beforeCancelCellEdit",
        "beforeCellDelete",
        "beforeCellEditStart",
        "beforeColumnDragStart",
        "beforeColumnDropFinalize",
        "beforeColumnResize",
        "beforeCopy",
        "beforeCSVExport",
        "beforeDependencyAdd",
        "beforeDependencyCreateDrag",
        "beforeDependencyCreateFinalize",
        "beforeDependencyDelete",
        "beforeDependencyEdit",
        "beforeDependencyEditShow",
        "beforeDependencySave",
        "beforeDestroy",
        "beforeDragCreate",
        "beforeDragCreateFinalize",
        "beforeEventAdd",
        "beforeEventDelete",
        "beforeEventDrag",
        "beforeEventDragSelect",
        "beforeEventDropFinalize",
        "beforeEventEdit",
        "beforeEventEditShow",
        "beforeEventResize",
        "beforeEventResizeFinalize",
        "beforeEventSave",
        "beforeEventSegmentDrag",
        "beforeEventSegmentDropFinalize",
        "beforeEventSegmentResize",
        "beforeEventSegmentResizeFinalize",
        "beforeEventSelectionChange",
        "beforeExcelExport",
        "beforeFillHandleDragStart",
        "beforeFinishCellEdit",
        "beforeHide",
        "beforePan",
        "beforePaste",
        "beforePdfExport",
        "beforePresetChange",
        "beforeRenderRow",
        "beforeRenderRows",
        "beforeRowCollapse",
        "beforeRowExpand",
        "beforeSelectionChange",
        "beforeSetRecord",
        "beforeShow",
        "beforeShowTerminals",
        "beforeTaskDelete",
        "beforeTaskEdit",
        "beforeTaskEditShow",
        "beforeTaskSave",
        "beforeToggleGroup",
        "beforeToggleNode",
        "cancelCellEdit",
        "catchAll",
        "cellClick",
        "cellContextMenu",
        "cellDblClick",
        "cellMenuBeforeShow",
        "cellMenuItem",
        "cellMenuShow",
        "cellMenuToggleItem",
        "cellMouseEnter",
        "cellMouseLeave",
        "cellMouseOut",
        "cellMouseOver",
        "collapse",
        "collapseNode",
        "columnDrag",
        "columnDragStart",
        "columnDrop",
        "columnResize",
        "columnResizeStart",
        "contextMenuItem",
        "contextMenuToggleItem",
        "copy",
        "currentTimelineUpdate",
        "dataChange",
        "dateRangeChange",
        "dependenciesDrawn",
        "dependencyClick",
        "dependencyContextMenu",
        "dependencyCreateDragStart",
        "dependencyCreateDrop",
        "dependencyDblClick",
        "dependencyMouseOut",
        "dependencyMouseOver",
        "dependencyValidationComplete",
        "dependencyValidationStart",
        "destroy",
        "dragCreateEnd",
        "dragCreateStart",
        "dragSelecting",
        "eventAutoCreated",
        "eventClick",
        "eventContextMenu",
        "eventDblClick",
        "eventDrag",
        "eventDragAbort",
        "eventDragModeChange",
        "eventDragReset",
        "eventDragStart",
        "eventDrop",
        "eventEditBeforeSetRecord",
        "eventKeyDown",
        "eventKeyUp",
        "eventMenuBeforeShow",
        "eventMenuItem",
        "eventMenuShow",
        "eventMouseDown",
        "eventMouseEnter",
        "eventMouseLeave",
        "eventMouseOut",
        "eventMouseOver",
        "eventMouseUp",
        "eventPartialResize",
        "eventResizeEnd",
        "eventResizeStart",
        "eventSegmentDrag",
        "eventSegmentDragAbort",
        "eventSegmentDragReset",
        "eventSegmentDragStart",
        "eventSegmentDrop",
        "eventSegmentPartialResize",
        "eventSegmentResizeEnd",
        "eventSegmentResizeStart",
        "eventSelectionChange",
        "expand",
        "expandNode",
        "fileDrop",
        "fillHandleBeforeDragFinalize",
        "fillHandleDrag",
        "fillHandleDragAbort",
        "fillHandleDragEnd",
        "fillHandleDragStart",
        "finishCellEdit",
        "focusIn",
        "focusOut",
        "gridRowBeforeDragStart",
        "gridRowBeforeDropFinalize",
        "gridRowDrag",
        "gridRowDragAbort",
        "gridRowDragStart",
        "gridRowDrop",
        "headerClick",
        "headerMenuBeforeShow",
        "headerMenuItem",
        "headerMenuShow",
        "headerMenuToggleItem",
        "hide",
        "lockRows",
        "mouseOut",
        "mouseOver",
        "navigate",
        "paint",
        "paste",
        "pasteNotAllowed",
        "pdfExport",
        "percentBarDrag",
        "percentBarDragAbort",
        "percentBarDragStart",
        "percentBarDrop",
        "presetChange",
        "recompose",
        "releaseEvent",
        "renderEvent",
        "renderRow",
        "renderRows",
        "resize",
        "resourceHeaderClick",
        "resourceHeaderContextmenu",
        "resourceHeaderDblclick",
        "resourceMenuBeforeShow",
        "resourceMenuItem",
        "resourceMenuShow",
        "resourceMenuToggleItem",
        "resourceNonWorkingTimeClick",
        "resourceNonWorkingTimeContextMenu",
        "resourceNonWorkingTimeDblClick",
        "resourceNonWorkingTimeMouseDown",
        "resourceNonWorkingTimeMouseOut",
        "resourceNonWorkingTimeMouseOver",
        "resourceNonWorkingTimeMouseUp",
        "resourceTimeRangeClick",
        "resourceTimeRangeContextMenu",
        "resourceTimeRangeDblClick",
        "resourceTimeRangeMouseDown",
        "resourceTimeRangeMouseOut",
        "resourceTimeRangeMouseOver",
        "resourceTimeRangeMouseUp",
        "responsive",
        "rowCollapse",
        "rowExpand",
        "rowMouseEnter",
        "rowMouseLeave",
        "scheduleClick",
        "scheduleContextMenu",
        "scheduleDblClick",
        "scheduleMenuBeforeShow",
        "scheduleMenuItem",
        "scheduleMenuShow",
        "scheduleMouseDown",
        "scheduleMouseEnter",
        "scheduleMouseLeave",
        "scheduleMouseMove",
        "scheduleMouseUp",
        "scroll",
        "scrollButtonClick",
        "selectionChange",
        "selectionModeChange",
        "show",
        "split",
        "splitterCollapseClick",
        "splitterDragEnd",
        "splitterDragStart",
        "splitterExpandClick",
        "startCellEdit",
        "subGridCollapse",
        "subGridExpand",
        "taskEditCanceled",
        "timeAxisChange",
        "timeAxisHeaderClick",
        "timeAxisHeaderContextMenu",
        "timeAxisHeaderDblClick",
        "timeAxisHeaderMenuBeforeShow",
        "timeAxisHeaderMenuItem",
        "timeAxisHeaderMenuShow",
        "timelineContextChange",
        "timelineViewportResize",
        "timeRangeHeaderClick",
        "timeRangeHeaderContextMenu",
        "timeRangeHeaderDblClick",
        "timeSelectionChange",
        "timeSelectionElementClick",
        "toggleGroup",
        "toggleNode",
        "toolClick",
        "transactionChange",
        "unlockRows",
        "unsplit",
        "visibleDateRangeChange"
      ]
    }, o = {}, i = u(), l = u(/* @__PURE__ */ new Map());
    return D("teleports", l), h(() => {
      const e = E();
      o.value = r.createWidget({
        me: e,
        props: v,
        emit: C,
        widgetData: s,
        element: i.value,
        processCellContent: r.processCellContent,
        hasFrameworkRenderer: r.hasFrameworkRenderer,
        toRaw: R
      }), r.relayStores(e.data, o.value);
      const t = (n, f) => x(() => e.props[n], f);
      r.watchProps(o.value, s, t);
    }), b(() => {
      var e, t;
      (t = (e = o.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), m({
      instance: o,
      refElement: i,
      teleports: l
    }), (e, t) => (a(), c("div", {
      ref_key: "refElement",
      ref: i
    }, [
      F("div", z, [
        (a(!0), c(M, null, w(l.value.values(), (n) => (a(), d(T, {
          key: n.key,
          to: n.to
        }, [
          (a(), d(k(n.is), y({
            ref_for: !0,
            ref: n.key,
            "data-key": n.key,
            "data-generation": n.generation
          }, n.bind, { "data-vue": "" }), null, 16, ["data-key", "data-generation"]))
        ], 8, ["to"]))), 128))
      ], 512)
    ], 512));
  }
});
export {
  P as _
};
