// @bryntum/schedulerpro-vue-3/lib/chunks/Dz7YQC-b.js
import { defineComponent as f, ref as l, provide as b, onMounted as y, getCurrentInstance as g, toRaw as x, onBeforeUnmount as C, openBlock as v, createElementBlock as w, watch as k } from "vue";
import { TimePicker as E } from "@bryntum/schedulerpro";
import { WrapperHelper as r } from "../helper/WrapperHelper.js";
const z = /* @__PURE__ */ f({
  __name: "BryntumTimePicker",
  props: {
    adopt: {},
    align: {},
    alignSelf: {},
    anchor: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    autoUpdateRecord: {},
    bbar: {},
    bodyCls: {},
    bubbleEvents: {},
    callOnFunctions: {},
    centered: {},
    cls: {},
    collapsed: {},
    collapsible: {},
    config: {},
    constrainTo: {},
    content: {},
    contentElementCls: {},
    dataset: {},
    defaultBindProperty: {},
    defaultFocus: { type: Function },
    defaults: {},
    disabled: {},
    dock: {},
    draggable: {},
    extraData: {},
    flex: {},
    floating: {},
    footer: {},
    format: {},
    header: {},
    height: {},
    hidden: {},
    hideAnimation: {},
    hideWhenEmpty: {},
    html: { type: [String, Function, Array] },
    htmlCls: {},
    id: {},
    ignoreParentReadOnly: {},
    insertBefore: {},
    insertFirst: {},
    itemCls: {},
    items: {},
    keyMap: {},
    layout: {},
    layoutStyle: {},
    lazyItems: {},
    listeners: {},
    localeClass: {},
    localizable: {},
    localizableProperties: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    max: {},
    maxHeight: {},
    maximizeOnMobile: {},
    maxWidth: {},
    min: {},
    minHeight: {},
    minWidth: {},
    monitorResize: {},
    namedItems: {},
    owner: {},
    positioned: {},
    preventTooltipOnTouch: {},
    readOnly: {},
    record: {},
    relayStoreEvents: {},
    ripple: {},
    rootElement: {},
    scrollable: {},
    scrollAction: {},
    seconds: {},
    showAnimation: {},
    showTooltipWhenDisabled: {},
    stateful: {},
    statefulEvents: {},
    stateId: {},
    stateProvider: {},
    strictRecordMapping: {},
    strips: {},
    tab: {},
    tag: {},
    tbar: {},
    textAlign: {},
    textContent: {},
    title: {},
    tools: {},
    tooltip: {},
    trapFocus: {},
    type: {},
    ui: {},
    value: {},
    weight: {},
    width: {},
    x: {},
    y: {}
  },
  emits: ["beforeDestroy", "beforeHide", "beforeSetRecord", "beforeShow", "catchAll", "collapse", "destroy", "expand", "focusIn", "focusOut", "hide", "paint", "recompose", "resize", "show", "timeChange", "toolClick"],
  setup(s, { expose: c, emit: d }) {
    const p = s, m = d, n = {
      instanceClass: E,
      instanceName: "TimePicker",
      configNames: [
        "adopt",
        "align",
        "anchor",
        "ariaDescription",
        "ariaLabel",
        "autoUpdateRecord",
        "bbar",
        "bodyCls",
        "bubbleEvents",
        "callOnFunctions",
        "centered",
        "collapsible",
        "config",
        "constrainTo",
        "contentElementCls",
        "defaultBindProperty",
        "defaultFocus",
        "defaults",
        "dock",
        "draggable",
        "floating",
        "footer",
        "header",
        "hideAnimation",
        "hideWhenEmpty",
        "htmlCls",
        "ignoreParentReadOnly",
        "itemCls",
        "lazyItems",
        "listeners",
        "localeClass",
        "localizable",
        "localizableProperties",
        "maskDefaults",
        "masked",
        "monitorResize",
        "namedItems",
        "owner",
        "positioned",
        "preventTooltipOnTouch",
        "relayStoreEvents",
        "ripple",
        "rootElement",
        "scrollAction",
        "seconds",
        "showAnimation",
        "showTooltipWhenDisabled",
        "stateful",
        "statefulEvents",
        "stateId",
        "stateProvider",
        "strips",
        "tab",
        "tag",
        "tbar",
        "textAlign",
        "textContent",
        "trapFocus",
        "type",
        "ui",
        "weight"
      ],
      propertyConfigNames: [
        "alignSelf",
        "appendTo",
        "ariaLive",
        "cls",
        "collapsed",
        "content",
        "dataset",
        "disabled",
        "extraData",
        "flex",
        "format",
        "height",
        "hidden",
        "html",
        "id",
        "insertBefore",
        "insertFirst",
        "items",
        "keyMap",
        "layout",
        "layoutStyle",
        "margin",
        "max",
        "maxHeight",
        "maximizeOnMobile",
        "maxWidth",
        "min",
        "minHeight",
        "minWidth",
        "onBeforeDestroy",
        "onBeforeHide",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onCatchAll",
        "onCollapse",
        "onDestroy",
        "onExpand",
        "onFocusIn",
        "onFocusOut",
        "onHide",
        "onPaint",
        "onReadOnly",
        "onRecompose",
        "onResize",
        "onShow",
        "onTimeChange",
        "onToolClick",
        "readOnly",
        "record",
        "scrollable",
        "strictRecordMapping",
        "title",
        "tools",
        "tooltip",
        "value",
        "width",
        "x",
        "y"
      ],
      propertyNames: [
        "accessibility",
        "anchorSize",
        "initialValue",
        "isSettingValues",
        "isValid",
        "state",
        "values"
      ],
      eventNames: [
        "beforeDestroy",
        "beforeHide",
        "beforeSetRecord",
        "beforeShow",
        "catchAll",
        "collapse",
        "destroy",
        "expand",
        "focusIn",
        "focusOut",
        "hide",
        "paint",
        "recompose",
        "resize",
        "show",
        "timeChange",
        "toolClick"
      ]
    }, o = {}, a = l(), i = l(/* @__PURE__ */ new Map());
    return b("teleports", i), y(() => {
      const e = g();
      o.value = r.createWidget({
        me: e,
        props: p,
        emit: m,
        widgetData: n,
        element: a.value,
        toRaw: x
      });
      const t = (h, u) => k(() => e.props[h], u);
      r.watchProps(o.value, n, t);
    }), C(() => {
      var e, t;
      (t = (e = o.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), c({
      instance: o,
      refElement: a,
      teleports: i
    }), (e, t) => (v(), w("div", {
      ref_key: "refElement",
      ref: a
    }, null, 512));
  }
});
export {
  z as _
};
