// @bryntum/schedulerpro-vue-3/lib/chunks/R6pSWfeV.js
import { defineComponent as h, ref as l, provide as b, onMounted as y, getCurrentInstance as g, toRaw as C, onBeforeUnmount as w, openBlock as x, createElementBlock as v, watch as E } from "vue";
import { CalendarEditor as k } from "@bryntum/schedulerpro";
import { WrapperHelper as r } from "../helper/WrapperHelper.js";
const S = /* @__PURE__ */ h({
  __name: "BryntumCalendarEditor",
  props: {
    activeDate: {},
    adopt: {},
    align: {},
    alignSelf: {},
    anchor: {},
    appendTo: {},
    ariaDescription: {},
    ariaLabel: {},
    ariaLive: {},
    autoClose: {},
    autoShow: {},
    autoUpdateRecord: {},
    bbar: {},
    bodyCls: {},
    bubbleEvents: {},
    calendar: {},
    callOnFunctions: {},
    centered: {},
    closable: {},
    closeAction: {},
    cls: {},
    collapsed: {},
    collapsible: {},
    config: {},
    constrainTo: {},
    content: {},
    contentElementCls: {},
    dataset: {},
    defaultBindProperty: {},
    defaultFocus: { type: Function },
    defaults: {},
    defaultWorkingDayAvailability: {},
    defaultWorkingDays: {},
    disabled: {},
    dock: {},
    draggable: {},
    extraData: {},
    flex: {},
    floating: {},
    focusOnToFront: {},
    footer: {},
    forElement: {},
    header: {},
    height: {},
    hidden: {},
    hideAnimation: {},
    hideWhenEmpty: {},
    html: { type: [String, Function, Array] },
    htmlCls: {},
    id: {},
    ignoreParentReadOnly: {},
    insertBefore: {},
    insertFirst: {},
    itemCls: {},
    items: {},
    keyMap: {},
    layout: {},
    layoutStyle: {},
    lazyItems: {},
    listeners: {},
    localeClass: {},
    localizable: {},
    localizableProperties: {},
    margin: {},
    maskDefaults: {},
    masked: {},
    maxHeight: {},
    maximizable: {},
    maximized: {},
    maximizeOnMobile: {},
    maxWidth: {},
    minHeight: {},
    minWidth: {},
    modal: {},
    monitorResize: {},
    namedItems: {},
    owner: {},
    positioned: {},
    preventTooltipOnTouch: {},
    readOnly: {},
    record: {},
    relayStoreEvents: {},
    ripple: {},
    rootElement: {},
    scrollable: {},
    scrollAction: {},
    showAnimation: {},
    showOnClick: {},
    showTooltipWhenDisabled: {},
    stateful: {},
    statefulEvents: {},
    stateId: {},
    stateProvider: {},
    strictRecordMapping: {},
    strips: {},
    tab: {},
    tag: {},
    tbar: {},
    textAlign: {},
    textContent: {},
    title: {},
    tools: {},
    tooltip: {},
    trapFocus: {},
    type: {},
    ui: {},
    weight: {},
    width: {},
    x: {},
    y: {}
  },
  emits: ["beforeClose", "beforeDestroy", "beforeHide", "beforeSetRecord", "beforeShow", "catchAll", "collapse", "destroy", "expand", "focusIn", "focusOut", "hide", "paint", "recompose", "resize", "show", "toolClick"],
  setup(s, { expose: c, emit: d }) {
    const p = s, m = d, n = {
      instanceClass: k,
      instanceName: "CalendarEditor",
      configNames: [
        "activeDate",
        "adopt",
        "align",
        "anchor",
        "ariaDescription",
        "ariaLabel",
        "autoClose",
        "autoShow",
        "autoUpdateRecord",
        "bbar",
        "bodyCls",
        "bubbleEvents",
        "calendar",
        "callOnFunctions",
        "centered",
        "closable",
        "closeAction",
        "collapsible",
        "config",
        "constrainTo",
        "contentElementCls",
        "defaultBindProperty",
        "defaultFocus",
        "defaults",
        "defaultWorkingDayAvailability",
        "defaultWorkingDays",
        "dock",
        "draggable",
        "floating",
        "focusOnToFront",
        "footer",
        "forElement",
        "header",
        "hideAnimation",
        "hideWhenEmpty",
        "htmlCls",
        "ignoreParentReadOnly",
        "itemCls",
        "lazyItems",
        "listeners",
        "localeClass",
        "localizable",
        "localizableProperties",
        "maskDefaults",
        "masked",
        "maximizable",
        "modal",
        "monitorResize",
        "namedItems",
        "owner",
        "positioned",
        "preventTooltipOnTouch",
        "relayStoreEvents",
        "ripple",
        "rootElement",
        "scrollAction",
        "showAnimation",
        "showOnClick",
        "showTooltipWhenDisabled",
        "stateful",
        "statefulEvents",
        "stateId",
        "stateProvider",
        "strips",
        "tab",
        "tag",
        "tbar",
        "textAlign",
        "textContent",
        "trapFocus",
        "type",
        "ui",
        "weight"
      ],
      propertyConfigNames: [
        "alignSelf",
        "appendTo",
        "ariaLive",
        "cls",
        "collapsed",
        "content",
        "dataset",
        "disabled",
        "extraData",
        "flex",
        "height",
        "hidden",
        "html",
        "id",
        "insertBefore",
        "insertFirst",
        "items",
        "keyMap",
        "layout",
        "layoutStyle",
        "margin",
        "maxHeight",
        "maximized",
        "maximizeOnMobile",
        "maxWidth",
        "minHeight",
        "minWidth",
        "onBeforeClose",
        "onBeforeDestroy",
        "onBeforeHide",
        "onBeforeSetRecord",
        "onBeforeShow",
        "onCatchAll",
        "onCollapse",
        "onDestroy",
        "onExpand",
        "onFocusIn",
        "onFocusOut",
        "onHide",
        "onPaint",
        "onReadOnly",
        "onRecompose",
        "onResize",
        "onShow",
        "onToolClick",
        "readOnly",
        "record",
        "scrollable",
        "strictRecordMapping",
        "title",
        "tools",
        "tooltip",
        "width",
        "x",
        "y"
      ],
      propertyNames: [
        "accessibility",
        "anchorSize",
        "isSettingValues",
        "isValid",
        "state",
        "values"
      ],
      eventNames: [
        "beforeClose",
        "beforeDestroy",
        "beforeHide",
        "beforeSetRecord",
        "beforeShow",
        "catchAll",
        "collapse",
        "destroy",
        "expand",
        "focusIn",
        "focusOut",
        "hide",
        "paint",
        "recompose",
        "resize",
        "show",
        "toolClick"
      ]
    }, o = {}, a = l(), i = l(/* @__PURE__ */ new Map());
    return b("teleports", i), y(() => {
      const e = g();
      o.value = r.createWidget({
        me: e,
        props: p,
        emit: m,
        widgetData: n,
        element: a.value,
        toRaw: C
      });
      const t = (u, f) => E(() => e.props[u], f);
      r.watchProps(o.value, n, t);
    }), w(() => {
      var e, t;
      (t = (e = o.value) == null ? void 0 : e.destroy) == null || t.call(e);
    }), c({
      instance: o,
      refElement: a,
      teleports: i
    }), (e, t) => (x(), v("div", {
      ref_key: "refElement",
      ref: a
    }, null, 512));
  }
});
export {
  S as _
};
